import json
import matplotlib.pyplot as plt
import operator
import os
import sys
import typing
import csv
from collections import defaultdict
from scipy import stats
import numpy as np
import pandas as pd
from pathlib import Path

from Bio import SeqIO
 


sys.path.append(str(Path(os.path.dirname(os.getcwd())).parent))


from generate_summary_file import generate_summary
from analysis.input_testing_data.generate_input_testing_data_for_modules import generate_testing_data_for_ecoli_and_bacillus
from modules.main import run_modules
from modules.main import run_input_processing
from modules.user_IO import UserInputModule
from modules.main import EvaluationModule
from modules.run_summary import RunSummary
from modules import shared_functions_and_vars
from modules import models


def get_user_input(optimization_method: str,
                   is_ecoli_optimized: bool,
                   orf_sequence: typing.Optional[str] = None,
                   orf_sequence_file: typing.Optional[str] = None,
                   output_path: typing.Optional[str] = None,
                   optimization_cub_index: str = "CAI",
                   tuning_param: float = 0.5,
                   should_use_mrna_levels = True,
                   should_use_protein_abundance = False):
    default_user_inp_raw = generate_testing_data_for_ecoli_and_bacillus(
        orf_optimization_method=optimization_method,
        orf_optimization_cub_index=optimization_cub_index,
        clusters_count=1,
        tuning_param=tuning_param,
        is_ecoli_optimized=is_ecoli_optimized,
        sequence=orf_sequence,
        sequence_file_path=orf_sequence_file,
        output_path=os.path.join("results", output_path),
        should_use_mrna_levels=should_use_mrna_levels,
        should_use_protein_abundance=should_use_protein_abundance,
    )
    
    # return run_input_processing(default_user_inp_raw) 
    return UserInputModule.run_module(default_user_inp_raw, initiation_optimized_codons_num=15, run_summary=RunSummary()) 


def evaluate_score_for_sequence(sequence: str,
                                user_input: models.ModuleInput,
                                optimization_method: str,
                                is_ecoli_optimized: bool,
                                orf_sequence: typing.Optional[str] = None,
                                orf_sequence_file: typing.Optional[str] = None,
                                output_path: typing.Optional[str] = None,
                                optimization_cub_index: str = "CAI",
                                tuning_param: float = 0.5,
                                should_use_mrna_levels = True,
                                should_use_protein_abundance = False):
    run_summary = RunSummary()
    return EvaluationModule.run_module(final_sequence=sequence, 
                                       module_input=user_input,
                                       optimization_cub_index=user_input.orf_optimization_cub_index,
                                       skipped_codons_num=15,
                                       run_summary=run_summary)


def run_single_method_for_orf_sequence(optimization_method: str,
                                       is_ecoli_optimized: bool,
                                       orf_sequence: typing.Optional[str] = None,
                                       orf_sequence_file: typing.Optional[str] = None,
                                       output_path: typing.Optional[str] = None,
                                       optimization_cub_index: str = "CAI",
                                       tuning_param: float = 0.5,
                                       should_use_mrna_levels = True,
                                       should_use_protein_abundance = False,
                                       initiation_optimization_method = "original"):
    default_user_inp_raw = generate_testing_data_for_ecoli_and_bacillus(
        orf_optimization_method=optimization_method,
        orf_optimization_cub_index=optimization_cub_index,
        clusters_count=1,
        tuning_param=tuning_param,
        is_ecoli_optimized=is_ecoli_optimized,
        sequence=orf_sequence,
        sequence_file_path=orf_sequence_file,
        output_path=os.path.join("results", output_path),
        should_use_mrna_levels=should_use_mrna_levels,
        should_use_protein_abundance=should_use_protein_abundance,
        initiation_optimization_method=initiation_optimization_method,
    )
    return run_modules(default_user_inp_raw, should_run_output_module=False)


def run_all_methods(orf_sequence = None, orf_sequence_file = None, output_path = None, tuning_param: int = 0.5, should_use_mrna_levels: bool = True, should_use_protein_abundance: bool = False, initiation_optimization_method: str = "original"):
    for optimization_method in [
        "single_codon_ratio",
        "single_codon_diff",
        # "single_codon_weakest_link",
        "zscore_single_aa_ratio",
        "zscore_bulk_aa_ratio",
        "zscore_single_aa_diff",
        "zscore_bulk_aa_diff",
        # "zscore_single_aa_weakest_link",
        # "zscore_bulk_aa_weakest_link",
    ]:
        for optimization_cub_index in [
            "CAI", 
            "tAI",
        ]:
            for direction in [True, False]:
                run_single_method_for_orf_sequence(optimization_method=optimization_method,
                                                   optimization_cub_index=optimization_cub_index,
                                                   is_ecoli_optimized=direction,
                                                   orf_sequence=orf_sequence,
                                                   orf_sequence_file=orf_sequence_file,
                                                   output_path=output_path,
                                                   tuning_param=tuning_param,
                                                   should_use_mrna_levels=should_use_mrna_levels,
                                                   should_use_protein_abundance=should_use_protein_abundance,
                                                   initiation_optimization_method=initiation_optimization_method)


def generate_sequences_fasta_file(root_dir) -> None:
    filename = "run_summary.json"

    sequences = []
    sequences_names = []
    for root, dirs, files in os.walk(root_dir):
        for file in files:
            if file == filename:
                directory_name = Path(root).name
                file_path = os.path.join(root, file)
                with open(file_path, "r") as summary_file:
                    results_json = json.load(summary_file)

                seq = results_json["final_evaluation"]["final_sequence"]
                sequences.append(seq)
                sequences_names.append(directory_name[:-5])
    
    file_name = os.path.join(root_dir, "mcherry_variants")
    shared_functions_and_vars.write_fasta(file_name, sequences, sequences_names)
    
    fields = ['variant', 'sequence']
    variant_dict = [{"variant": sequences_names[i], "sequence": sequences[i]} for i in range(len(sequences))]
    with open(file_name+".csv", 'w') as csvfile:
        writer = csv.DictWriter(csvfile, fieldnames=fields)
        writer.writeheader()
        writer.writerows(variant_dict)


def calculate_codon_frequencies(referece_cds: typing.Sequence[str]):
    codons_counter = defaultdict(int)
    
    for cds in referece_cds:
        if len(cds) % 3 != 0:
            continue
        for i in range(0, len(cds), 3):
            codon = cds[i:i+3]
            codons_counter[codon] += 1
    
    # Calculate relative frequncies
    for amino_acid, codons in shared_functions_and_vars.synonymous_codons.items():
        total_amino_acid_codons = 0
        for amino_acid_codon in codons:
            total_amino_acid_codons += codons_counter[amino_acid_codon]
        
        # In case a certain amino acid is missing from the reference cds collection, no need to normalize
        if total_amino_acid_codons == 0:
            continue
        
        for amino_acid_codon in codons:
            codons_counter[amino_acid_codon] /= total_amino_acid_codons
    
    codons = list(codons_counter.keys())
    df_dict = {"codon": codons, "amino_acid": [shared_functions_and_vars.nt_to_aa[codon] for codon in codons], "frequency": [codons_counter[codon] for codon in codons]}

    return pd.DataFrame(df_dict)


def calculate_codon_frequencies_lamda(referece_cds: typing.Sequence[str]):
    codons_counter = defaultdict(int)
    
    for cds in referece_cds:
        if len(cds) % 3 != 0:
            continue
        for i in range(0, len(cds), 3):
            codon = cds[i:i+3]
            codons_counter[codon] += 1
    
    # Calculate relative frequncies
    for amino_acid, codons in shared_functions_and_vars.synonymous_codons.items():
        amino_acid_codon_to_count = {codon: codons_counter[codon] for codon in codons}
        most_frequent_codon = max(amino_acid_codon_to_count, key=amino_acid_codon_to_count.get)
        
        if amino_acid_codon_to_count[most_frequent_codon] == 0:
            # In case a certain amino acid is missing from the reference cds collection, no need to normalize
            continue
            
        for amino_acid_codon in codons:
            codons_counter[amino_acid_codon] = 1 if amino_acid_codon == most_frequent_codon else 0 
    
    codons = list(codons_counter.keys())
    df_dict = {"codon": codons, "amino_acid": [shared_functions_and_vars.nt_to_aa[codon] for codon in codons], "frequency": [codons_counter[codon] for codon in codons]}
    
    return pd.DataFrame(df_dict)


def plot_frequency_map(frequency_df):
    sorted_df = frequency_df.sort_values("amino_acid")
    
    plt.rcParams["figure.figsize"] = [10, 5]
    plt.rcParams["figure.autolayout"] = True

    sorted_df.plot("codon", "frequency", kind="bar", color = "b")


def calculate_distance_between_frequencies(first_codons_freq, second_codons_freq):
    return np.linalg.norm(first_codons_freq-second_codons_freq)


def get_variants_df_dict(variants_root_path, ecoli_codon_frequencies, bacillus_codon_frequencies, ecoli_codon_frequencies_lamda, bacillus_codon_frequencies_lamda):
    with open(variants_root_path, "r") as mcherry_variants_fasta_file:
        mcherry_variants_dict = SeqIO.to_dict(SeqIO.parse(mcherry_variants_fasta_file, "fasta"))

    unified_variants_dict = defaultdict(dict)

    for variant_name, variant_seq in mcherry_variants_dict.items():  
        stripped_name = variant_name.replace("_False", "").replace("_True", "").replace("_ecoli_opt", "")
       
        mcherry_variant_codon_frequencies = calculate_codon_frequencies((str(variant_seq.seq),)).sort_values("codon")
        mcherry_variant_codon_frequencies_lamda = calculate_codon_frequencies_lamda((str(variant_seq.seq),)).sort_values("codon")

        series1 = mcherry_variant_codon_frequencies["frequency"].reset_index(drop=True)
        series2 =  ecoli_codon_frequencies["frequency"].reset_index(drop=True)
        series3 =  bacillus_codon_frequencies["frequency"].reset_index(drop=True)
        
        ecoli_dist = calculate_distance_between_frequencies(series1, series2)
        bacillus_dist = calculate_distance_between_frequencies(series1, series3)
        
        series1_lamda = mcherry_variant_codon_frequencies_lamda["frequency"].reset_index(drop=True)
        series2_lamda =  ecoli_codon_frequencies_lamda["frequency"].reset_index(drop=True)
        series3_lamda =  bacillus_codon_frequencies_lamda["frequency"].reset_index(drop=True)
        
        ecoli_dist_lamda = calculate_distance_between_frequencies(series1_lamda, series2_lamda)
        bacillus_dist_lamda = calculate_distance_between_frequencies(series1_lamda, series3_lamda)

        if variant_name.endswith("True"):
            unified_variants_dict[stripped_name]["ecoli_dist_for_optimized_ecoli"] = ecoli_dist
            unified_variants_dict[stripped_name]["bacillus_dist_for_optimized_ecoli"] = bacillus_dist
            unified_variants_dict[stripped_name]["ecoli_dist_lamda_for_optimized_ecoli"] = ecoli_dist_lamda
            unified_variants_dict[stripped_name]["bacillus_dist_lamda_for_optimized_ecoli"] = bacillus_dist_lamda
        else:
            unified_variants_dict[stripped_name]["ecoli_dist_for_optimized_bacillus"] = ecoli_dist
            unified_variants_dict[stripped_name]["bacillus_dist_for_optimized_bacillus"] = bacillus_dist
            unified_variants_dict[stripped_name]["ecoli_dist_lamda_for_optimized_bacillus"] = ecoli_dist_lamda
            unified_variants_dict[stripped_name]["bacillus_dist_lamda_for_optimized_bacillus"] = bacillus_dist_lamda

    df_dict = defaultdict(list)
    for variant_name in unified_variants_dict.keys():
        df_dict["variant"].append(variant_name)
        df_dict["ecoli_dist_for_original_mcherry"].append(original_ecoli_dist)
        df_dict["ecoli_dist_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["ecoli_dist_for_optimized_ecoli"])
        df_dict["ecoli_dist_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["ecoli_dist_for_optimized_bacillus"])
        df_dict["ecoli_dist_lamda_for_original_mcherry"].append(original_ecoli_dist_lamda)
        df_dict["ecoli_dist_lamda_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["ecoli_dist_lamda_for_optimized_ecoli"])
        df_dict["ecoli_dist_lamda_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["ecoli_dist_lamda_for_optimized_bacillus"])

        df_dict["bacillus_dist_for_original_mcherry"].append(original_bacillus_dist)
        df_dict["bacillus_dist_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["bacillus_dist_for_optimized_ecoli"])
        df_dict["bacillus_dist_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["bacillus_dist_for_optimized_bacillus"])
        df_dict["bacillus_dist_lamda_for_original_mcherry"].append(original_bacillus_dist_lamda)
        df_dict["bacillus_dist_lamda_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["bacillus_dist_lamda_for_optimized_ecoli"])
        df_dict["bacillus_dist_lamda_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["bacillus_dist_lamda_for_optimized_bacillus"])
    
    return df_dict


# def get_variants_df_dict_highly_expressed(variants_root_path):
#     with open(variants_root_path, "r") as mcherry_variants_fasta_file:
#         mcherry_variants_dict = SeqIO.to_dict(SeqIO.parse(mcherry_variants_fasta_file, "fasta"))

#     unified_variants_dict = defaultdict(dict)

#     for variant_name, variant_seq in mcherry_variants_dict.items():
#         stripped_name = variant_name.replace("_False", "").replace("_True", "").replace("_ecoli_opt", "")
       
#         mcherry_variant_codon_frequencies = calculate_codon_frequencies((str(variant_seq.seq),)).sort_values("codon")
#         mcherry_variant_codon_frequencies_lamda = calculate_codon_frequencies_lamda((str(variant_seq.seq),)).sort_values("codon")

#         series1 = mcherry_variant_codon_frequencies["frequency"].reset_index(drop=True)
#         series2 =  ecoli_highly_expressed_genes_codon_frequencies["frequency"].reset_index(drop=True)
#         series3 =  bacillus_highly_expressed_genes_codon_frequencies["frequency"].reset_index(drop=True)

#         ecoli_dist = calculate_distance_between_frequencies(series1, series2)
#         bacillus_dist = calculate_distance_between_frequencies(series1, series3)
        
#         series1_lamda = mcherry_variant_codon_frequencies_lamda["frequency"].reset_index(drop=True)
#         series2_lamda =  ecoli_highly_expressed_genes_codon_frequencies_lamda["frequency"].reset_index(drop=True)
#         series3_lamda =  bacillus_highly_expressed_genes_codon_frequencies_lamda["frequency"].reset_index(drop=True)

#         ecoli_dist_lamda = calculate_distance_between_frequencies(series1_lamda, series2_lamda)
#         bacillus_dist_lamda = calculate_distance_between_frequencies(series1_lamda, series3_lamda)

#         if variant_name.endswith("True"):
#             unified_variants_dict[stripped_name]["ecoli_dist_for_optimized_ecoli"] = ecoli_dist
#             unified_variants_dict[stripped_name]["bacillus_dist_for_optimized_ecoli"] = bacillus_dist
#             unified_variants_dict[stripped_name]["ecoli_dist_lamda_for_optimized_ecoli"] = ecoli_dist_lamda
#             unified_variants_dict[stripped_name]["bacillus_dist_lamda_for_optimized_ecoli"] = bacillus_dist_lamda
#         else:
#             unified_variants_dict[stripped_name]["ecoli_dist_for_optimized_bacillus"] = ecoli_dist
#             unified_variants_dict[stripped_name]["bacillus_dist_for_optimized_bacillus"] = bacillus_dist
#             unified_variants_dict[stripped_name]["ecoli_dist_lamda_for_optimized_bacillus"] = ecoli_dist_lamda
#             unified_variants_dict[stripped_name]["bacillus_dist_lamda_for_optimized_bacillus"] = bacillus_dist_lamda

#     df_dict = defaultdict(list)
#     for variant_name in unified_variants_dict.keys():
#         df_dict["variant"].append(variant_name)
#         df_dict["ecoli_dist_for_original_mcherry"].append(original_ecoli_dist)
#         df_dict["ecoli_dist_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["ecoli_dist_for_optimized_ecoli"])
#         df_dict["ecoli_dist_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["ecoli_dist_for_optimized_bacillus"])
#         df_dict["ecoli_dist_lamda_for_original_mcherry"].append(original_ecoli_dist_lamda)
#         df_dict["ecoli_dist_lamda_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["ecoli_dist_lamda_for_optimized_ecoli"])
#         df_dict["ecoli_dist_lamda_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["ecoli_dist_lamda_for_optimized_bacillus"])

#         df_dict["bacillus_dist_for_original_mcherry"].append(original_bacillus_dist)
#         df_dict["bacillus_dist_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["bacillus_dist_for_optimized_ecoli"])
#         df_dict["bacillus_dist_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["bacillus_dist_for_optimized_bacillus"])
#         df_dict["bacillus_dist_lamda_for_original_mcherry"].append(original_bacillus_dist_lamda)
#         df_dict["bacillus_dist_lamda_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["bacillus_dist_lamda_for_optimized_ecoli"])
#         df_dict["bacillus_dist_lamda_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["bacillus_dist_lamda_for_optimized_bacillus"])
    
#     return df_dict


def get_codon_frequencies_df_dict(variants_root_path):
    with open(variants_root_path, "r") as mcherry_variants_fasta_file:
        mcherry_variants_dict = SeqIO.to_dict(SeqIO.parse(mcherry_variants_fasta_file, "fasta"))

    unified_variants_dict = defaultdict(dict)

    for variant_name, variant_seq in mcherry_variants_dict.items():
        if "weakest_link" in variant_name:
            continue
        
        stripped_name = variant_name.replace("_False", "").replace("_True", "").replace("_ecoli_opt", "")
       
        mcherry_variant_codon_frequencies = calculate_codon_frequencies((str(variant_seq.seq),)).sort_values("codon")
        
        return mcherry_variant_codon_frequencies 
    
        series1 = mcherry_variant_codon_frequencies["frequency"].reset_index(drop=True)
        
        if variant_name.endswith("True"):
            unified_variants_dict[stripped_name]["codon_frequencies_for_optimized_ecoli"] = mcherry_variant_codon_frequencies
        else:
            unified_variants_dict[stripped_name]["codon_frequencies_for_optimized_bacillus"] = mcherry_variant_codon_frequencies

    df_dict = defaultdict(list)
    for variant_name in unified_variants_dict.keys():
        df_dict["variant"].append(variant_name)
        df_dict["codon_frequencies_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["codon_frequencies_for_optimized_ecoli"])
        df_dict["codon_frequencies_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["codon_frequencies_for_optimized_bacillus"])
    
    return df_dict


def get_evaluation_summary(summary: typing.Dict[str, typing.Any], evaluation_method: str = "average_distance_score") -> typing.Dict[str, typing.Any]:
    final_evaluation = summary["final_evaluation"]
    for evaluation_summary in summary["evaluation"]:
        if evaluation_summary[evaluation_method] == final_evaluation[evaluation_method]:
            return evaluation_summary
    raise RuntimeError(F"Did not find a final evaluation for: {summary}")


def get_cub_score_df_dict(root_dir, evaluation_method: str = "average_distance_score"):
    filename = "run_summary.json"
    
    unified_variants_dict = defaultdict(dict)
    
    for root, dirs, files in os.walk(root_dir):
        for file in files:
            if file == filename:
                directory_name = Path(root).name
                file_path = os.path.join(root, file)
                with open(file_path, "r") as summary_file:
                    results_json = json.load(summary_file)
                
                is_cai = directory_name.startswith("CAI")
                evaluation_summary = get_evaluation_summary(results_json)
                organisms = evaluation_summary["organisms"]
                for organism in organisms:
                    if organism["name"] == "Bacillus subtilis":
                        if is_cai:
                            bacillus_initial_score = organism["cai_initial_score"]
                            bacillus_final_score = organism["cai_final_score"]
                        else:
                            bacillus_initial_score = organism["tai_initial_score"]
                            bacillus_final_score = organism["tai_final_score"]
                    else:
                        if is_cai:
                            ecoli_initial_score = organism["cai_initial_score"]
                            ecoli_final_score = organism["cai_final_score"]
                        else:
                            ecoli_initial_score = organism["tai_initial_score"]
                            ecoli_final_score = organism["tai_final_score"]
                            
                
                variant_name = directory_name[:-5]
                stripped_name = variant_name.replace("_False", "").replace("_True", "").replace("_ecoli_opt", "")
                
                if "weakest_link" in variant_name:
                    continue
                
                if variant_name.endswith("True"):
                    suffix = "_for_optimized_ecoli"
                else:
                    suffix = "_for_optimized_bacillus"
                
                unified_variants_dict[stripped_name]["bacillus_initial_score" + suffix] = bacillus_initial_score
                unified_variants_dict[stripped_name]["bacillus_final_score" + suffix] = bacillus_final_score
                unified_variants_dict[stripped_name]["ecoli_initial_score" + suffix] = ecoli_initial_score
                unified_variants_dict[stripped_name]["ecoli_final_score" + suffix] = ecoli_final_score
                unified_variants_dict[stripped_name]["average_distance_evaluation_score" + suffix] = evaluation_summary["average_distance_score"]
                unified_variants_dict[stripped_name]["weakest_link_evaluation_score" + suffix] = evaluation_summary["weakest_link_score"]
                unified_variants_dict[stripped_name]["ratio_evaluation_score" + suffix] = evaluation_summary["ratio_score"]
                
    
    df_dict = defaultdict(list)
    for variant_name in unified_variants_dict.keys():
        df_dict["variant"].append(variant_name)
        df_dict["ecoli_initial_score_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["ecoli_initial_score_for_optimized_ecoli"])
        df_dict["ecoli_final_score_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["ecoli_final_score_for_optimized_ecoli"])
        df_dict["ecoli_initial_score_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["ecoli_initial_score_for_optimized_bacillus"])
        df_dict["ecoli_final_score_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["ecoli_final_score_for_optimized_bacillus"])
        df_dict["bacillus_initial_score_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["bacillus_initial_score_for_optimized_ecoli"])
        df_dict["bacillus_final_score_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["bacillus_final_score_for_optimized_ecoli"])
        df_dict["bacillus_initial_score_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["bacillus_initial_score_for_optimized_bacillus"])
        df_dict["bacillus_final_score_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["bacillus_final_score_for_optimized_bacillus"])
        df_dict["average_distance_evaluation_score_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["average_distance_evaluation_score_for_optimized_ecoli"])
        df_dict["average_distance_evaluation_score_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["average_distance_evaluation_score_for_optimized_bacillus"])
        df_dict["weakest_link_evaluation_score_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["weakest_link_evaluation_score_for_optimized_ecoli"])
        df_dict["weakest_link_evaluation_score_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["weakest_link_evaluation_score_for_optimized_bacillus"])
        df_dict["ratio_evaluation_score_for_optimized_ecoli"].append(unified_variants_dict[variant_name]["ratio_evaluation_score_for_optimized_ecoli"])
        df_dict["ratio_evaluation_score_for_optimized_bacillus"].append(unified_variants_dict[variant_name]["ratio_evaluation_score_for_optimized_bacillus"])
    
    return df_dict


def get_cub_weights(organism_summary_file_path: str):
    with open(organism_summary_file_path, "r") as organism_summary_file:
        organism_summary = json.load(organism_summary_file)
    
    cai_profile = organism_summary["cai_weights"]
    tai_profile = organism_summary["tai_weights"]
    
    return cai_profile, tai_profile


def get_cub_scores(organism_summary_file_path: str):
    with open(organism_summary_file_path, "r") as organism_summary_file:
        organism_summary = json.load(organism_summary_file)
    
    cai_scores = organism_summary["cai_scores"]
    tai_scores = organism_summary["tai_scores"]
    
    return cai_scores, tai_scores


def get_mrna_estimated_expression(endogenous_genes_fasta_file_path, mrna_levels_file_path):
    expression_df = pd.read_csv(mrna_levels_file_path)
    gene_name_to_mrna_level = {}
    for _, pair in enumerate(zip(expression_df.gene.to_list(), expression_df.mRNA_level.to_list())):
        measured_gene_name, expression_level = pair
        try:
            gene_name_to_mrna_level[measured_gene_name.lower()] = float(expression_level)
        except:
            continue
            
    with open(endogenous_genes_fasta_file_path, "r") as endogenous_genes_fasta_file:
        endogenous_genes_dict = SeqIO.to_dict(SeqIO.parse(endogenous_genes_fasta_file, "fasta"), lambda x: x.description)
        
    estimated_expression = {description: {"cds": cds, "mrna_level": gene_name_to_mrna_level[description.split("|")[0].lower()]}  for description, cds in endogenous_genes_dict.items() if description.split("|")[0].lower() in gene_name_to_mrna_level}
    return estimated_expression


def get_pa_estimated_expression(endogenous_genes_fasta_file_path, pa_file_path):
    expression_df = pd.read_json(pa_file_path)
    gene_name_to_pa = {}
    for _, pair in enumerate(zip(expression_df.name.to_list(), expression_df.abundance.to_list())):
        measured_gene_name, expression_level = pair
        try:
            gene_name_to_pa[measured_gene_name.lower()] = float(expression_level)
        except:
            continue
            
    with open(endogenous_genes_fasta_file_path, "r") as endogenous_genes_fasta_file:
        endogenous_genes_dict = SeqIO.to_dict(SeqIO.parse(endogenous_genes_fasta_file, "fasta"), lambda x: x.description)
        
    estimated_expression = {description: {"cds": cds, "pa": gene_name_to_pa[description.split("|")[0].lower()]}  for description, cds in endogenous_genes_dict.items() if description.split("|")[0].lower() in gene_name_to_pa}
    return estimated_expression


def get_estimated_expression(endogenous_genes_fasta_file_path, expression_levels_file_path, expression_type: str):
    if expression_type == "mrna_levels":
        return get_mrna_estimated_expression(endogenous_genes_fasta_file_path, expression_levels_file_path)
    elif expression_type == "protein_abundance":
        return get_pa_estimated_expression(endogenous_genes_fasta_file_path, expression_levels_file_path)
    raise ValueError("Missing expression type support")


def get_highly_expressed_genes(endogenous_genes_fasta_file_path, expression_levels_file_path, expression_type):
    estimated_expression = get_estimated_expression(endogenous_genes_fasta_file_path=endogenous_genes_fasta_file_path, expression_levels_file_path=expression_levels_file_path, expression_type=expression_type)
    estimated_expression_threshold = 0.15
    
    expression_key = "mrna_level" if expression_type == "mrna_levels" else "pa"
    sorted_estimated_expression = dict(
        sorted(estimated_expression.items(), key=lambda x: x[1][expression_key], reverse=True)
    )
    highly_expressed_genes_count = round(len(sorted_estimated_expression) * estimated_expression_threshold)
    
    highly_expressed_names = list(sorted_estimated_expression.keys())[:highly_expressed_genes_count]
    highly_expressed_genes = {key: value for key, value in estimated_expression.items() if key in highly_expressed_names}
    return highly_expressed_genes


base_path = os.path.join(Path(os.getcwd()).parent.resolve(), "example_data")
macherry_file_path = os.path.join(base_path, "mCherry_original.fasta")


with open(macherry_file_path, "r") as mcherry_fasta_file:
    mcherry_dict = SeqIO.to_dict(SeqIO.parse(mcherry_fasta_file, "fasta"))

original_mcherry_seq = list(mcherry_dict.values())[0].seq











initiation_optimization_method="external"
mcherry_output_path = f"mcherry-wanted-constraint-init-{initiation_optimization_method}-0.1"
run_all_methods(orf_sequence_file=macherry_file_path, output_path=mcherry_output_path, should_use_protein_abundance=True, should_use_mrna_levels=False, initiation_optimization_method=initiation_optimization_method)


mcherry_root_dir = os.path.join(os.getcwd(), "results", mcherry_output_path)
mcherry_variants_file = os.path.join(mcherry_root_dir, "mcherry_variants.fasta")
generate_sequences_fasta_file(mcherry_root_dir)
generate_summary(results_directory=mcherry_root_dir)

reference_mcherry_protein_seq  = "MVSKGEEDNMAIIKEFMRFKVHMEGSVNGHEFEIEGEGEGRPYEGTQTAKLKVTKGGPLPFAWDILSPQFMYGSKAYVKHPADIPDYLKLSFPEGFKWERVMNFEDGGVVTVTQDSSLQDGEFIYKVKLRGTNFPSDGPVMQKKTMGWEASSERMYPEDGALKGEIKQRLKLKDGGHYDAEVKTTYKAKKPVQLPGAYNVNIKLDITSHNEDYTIVEQYERAEGRHSTGGMDELYK*"
formatted_reference_mcherry_protein_seq = reference_mcherry_protein_seq.replace("*", "_")

with open(os.path.join(mcherry_root_dir, "mcherry_variants.fasta"), "r") as fasta_file:
     mcherry_dict = SeqIO.to_dict(SeqIO.parse(fasta_file, "fasta"))
     for seq_key, seq in mcherry_dict.items():
         translated_variant_mcherry_seq = shared_functions_and_vars.translate(str(seq.seq))
     
         assert translated_variant_mcherry_seq == formatted_reference_mcherry_protein_seq
         
         seqeunce_str = str(seq.seq)
         seqeunce_str_split = [seqeunce_str[i:i+3].upper() for i in range(0, len(seqeunce_str), 3)]
         for i in range(15, len(seqeunce_str_split)):
            prev_codon = seqeunce_str_split[i-1]
            codon = seqeunce_str_split[i]
            if prev_codon == codon:
                print(f"codon {codon} is the same as it previous, location {i} in {seq_key}")
     
     variant_seqs = [str(seq.seq) for seq in mcherry_dict.values()] 
     assert len(variant_seqs) == len(set(variant_seqs)) == 16
    
     variant_seqs_init = [seq[:15*3] for seq in variant_seqs]
     assert len(set(variant_seqs_init)) == 1
    
     


import codonbias as cb
nuc = cb.stats.BaseCounter()
for variant_name, variant_seq in mcherry_dict.items():
    print("checking " + variant_name)
    freq = nuc.count(variant_seq).get_table(normed=True)
    gc_content = freq['G'] + freq['C']
    print(f"GC content: {gc_content}")
    prev_codon = None
    repeating_seq_len = 0
    repeating_index = None
    variant_seq = str(variant_seq)
    for i in range(0, len(variant_seq), 3):
        codon = variant_seq[i:i+3]
        if codon == prev_codon:
            repeating_seq_len += 1
            if repeating_index is None:
                repeating_index = i
        else:
            if repeating_seq_len != 0:
                print(f"found repeating of codon {prev_codon} of length {repeating_seq_len} starting at index {repeating_index}")
            repeating_seq_len = 0
            repeating_index = None
        prev_codon = codon








run_all_methods(orf_sequence_file=macherry_file_path, output_path="mcherry_greater_alpha", tuning_param=1, should_use_protein_abundance=True, should_use_mrna_levels=False,  initiation_optimization_method=initiation_optimization_method)


mcherry_greater_alpha_root_dir = r"C:\projects\Igem_TAU_2021_moran\analysis\orf_model_analysis\results\mcherry_greater_alpha"
mcherry_greater_alpha_variants_file = os.path.join(mcherry_greater_alpha_root_dir, "mcherry_variants.fasta")
generate_sequences_fasta_file(mcherry_greater_alpha_root_dir)
generate_summary(results_directory=mcherry_greater_alpha_root_dir)








run_all_methods(orf_sequence_file=macherry_file_path, output_path="mcherry_lower_alpha", tuning_param=0, should_use_protein_abundance=True, should_use_mrna_levels=False,  initiation_optimization_method=initiation_optimization_method)


mcherry_lower_alpha_root_dir = r"C:\projects\Igem_TAU_2021_moran\analysis\orf_model_analysis\results\mcherry_lower_alpha"
mcherry_lower_alpha_variants_file = os.path.join(mcherry_lower_alpha_root_dir, "mcherry_variants.fasta")
generate_sequences_fasta_file(mcherry_lower_alpha_root_dir)
generate_summary(results_directory=mcherry_lower_alpha_root_dir)





# run_all_methods(orf_sequence_file=macherry_file_path, output_path="ribosomal", should_use_mrna_levels=False)


# mcherry_ribosomal_root_dir = r"C:\projects\Igem_TAU_2021_moran\analysis\orf_model_analysis\results\mrna\ribosomal"
# mcherry_ribosomal_variants_file = os.path.join(mcherry_ribosomal_root_dir, "mcherry_variants.fasta")
# generate_sequences_fasta_file(mcherry_ribosomal_root_dir)
# generate_summary(results_directory=mcherry_ribosomal_root_dir)





bacillus_endogenous_genes_fasta_file_path = "Bacillus subtilis.fasta"
ecoli_endogenous_genes_fasta_file_path =  "Escherichia coli.fasta"


with open(bacillus_endogenous_genes_fasta_file_path, "r") as bacillus_endogenous_genes_fasta_file:
    bacillus_endogenous_genes_dict = SeqIO.to_dict(SeqIO.parse(bacillus_endogenous_genes_fasta_file, "fasta"), lambda record: record.description)

bacillus_codon_frequencies = calculate_codon_frequencies([str(gene.seq) for gene in bacillus_endogenous_genes_dict.values()])
bacillus_codon_frequencies = bacillus_codon_frequencies.sort_values("codon")

bacillus_codon_frequencies_lamda = calculate_codon_frequencies_lamda([str(gene.seq) for gene in bacillus_endogenous_genes_dict.values()])
bacillus_codon_frequencies_lamda = bacillus_codon_frequencies_lamda.sort_values("codon")

with open(ecoli_endogenous_genes_fasta_file_path, "r") as ecoli_endogenous_genes_fasta_file:
    ecoli_endogenous_genes_dict = SeqIO.to_dict(SeqIO.parse(ecoli_endogenous_genes_fasta_file, "fasta"), lambda x: x.description)

ecoli_codon_frequencies = calculate_codon_frequencies([str(gene.seq) for gene in ecoli_endogenous_genes_dict.values()])
ecoli_codon_frequencies = ecoli_codon_frequencies.sort_values("codon")

ecoli_codon_frequencies_lamda = calculate_codon_frequencies_lamda([str(gene.seq) for gene in ecoli_endogenous_genes_dict.values()])
ecoli_codon_frequencies_lamda = ecoli_codon_frequencies_lamda.sort_values("codon")


bacillus_codon_frequencies


ecoli_codon_frequencies


ecoli_highly_expressed_genes = get_highly_expressed_genes(endogenous_genes_fasta_file_path=ecoli_endogenous_genes_fasta_file_path,
                                                          expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\PA_ecoli.json",
                                                          expression_type="protein_abundance")
# ecoli_highly_expressed_genes = get_highly_expressed_genes(endogenous_genes_fasta_file_path=ecoli_endogenous_genes_fasta_file_path,
#                                                           expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\ecoli_mrna_level.csv",
#                                                           expression_type="mrna_levels")
ecoli_highly_expressed_genes_codon_frequencies = calculate_codon_frequencies([str(gene["cds"].seq) for gene in ecoli_highly_expressed_genes.values()])
ecoli_highly_expressed_genes_codon_frequencies = ecoli_highly_expressed_genes_codon_frequencies.sort_values("codon")

ecoli_highly_expressed_genes_codon_frequencies_lamda = calculate_codon_frequencies_lamda([str(gene["cds"].seq) for gene in ecoli_highly_expressed_genes.values()])
ecoli_highly_expressed_genes_codon_frequencies_lamda = ecoli_highly_expressed_genes_codon_frequencies_lamda.sort_values("codon")

# bacillus_highly_expressed_genes = get_highly_expressed_genes(endogenous_genes_fasta_file_path=bacillus_endogenous_genes_fasta_file_path,
#                                                           expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\bacillus_mrna_level.csv",
#                                                             expression_type="mrna_levels")
bacillus_highly_expressed_genes = get_highly_expressed_genes(endogenous_genes_fasta_file_path=bacillus_endogenous_genes_fasta_file_path,
                                                             expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\PA_bacillus.json",
                                                            expression_type="protein_abundance")
bacillus_highly_expressed_genes_codon_frequencies = calculate_codon_frequencies([str(gene["cds"].seq) for gene in bacillus_highly_expressed_genes.values()])
bacillus_highly_expressed_genes_codon_frequencies = bacillus_highly_expressed_genes_codon_frequencies.sort_values("codon")

bacillus_highly_expressed_genes_codon_frequencies_lamda = calculate_codon_frequencies_lamda([str(gene["cds"].seq) for gene in bacillus_highly_expressed_genes.values()])
bacillus_highly_expressed_genes_codon_frequencies_lamda = bacillus_highly_expressed_genes_codon_frequencies_lamda.sort_values("codon")





from modules.user_IO.user_input import UserInputModule
from modules.run_summary import RunSummary

from generate_summary_file import generate_summary
from analysis.input_testing_data.generate_input_testing_data_for_modules import generate_testing_data_for_ecoli_and_bacillus
from modules.main import run_modules
from modules import shared_functions_and_vars

default_user_inp_raw = generate_testing_data_for_ecoli_and_bacillus(
    orf_optimization_method="single_codon_diff",
    orf_optimization_cub_index="CAI",
    clusters_count=1,
    tuning_param=0.5,
    is_ecoli_optimized=True,
    sequence=original_mcherry_seq,
    output_path="results",
    should_use_mrna_levels=False,
    should_use_protein_abundance=True,
)

UserInputModule.run_module(default_user_inp_raw, initiation_optimized_codons_num=15, run_summary=RunSummary())


ecoli_cai_weights, ecoli_tai_weights = get_cub_weights(r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\Escherichia-coli_True_parsed.json")
bacillus_cai_weights, bacillus_tai_weights = get_cub_weights(r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\Bacillus-subtilis_False_parsed.json")


codons = ecoli_cai_weights.keys()
weights_df_dict = {
    "codon": codons, 
    "amino_acid": [shared_functions_and_vars.nt_to_aa[k] for k in codons],
    "ecoli_cai_weights": [ecoli_cai_weights[k] for k in codons],
    # "ecoli_tai_weights": [ecoli_tai_weights.get(k, 0) for k in codons],
    "bacillus_cai_weights": [bacillus_cai_weights[k] for k in codons],
    #"bacillus_tai_weights": [bacillus_tai_weights.get(k, 0) for k in codons],
}
weights_dataframe = pd.DataFrame(weights_df_dict)
weights_dataframe.sort_values("amino_acid")
weights_dataframe


fig, axes = plt.subplots(nrows=1,ncols=2, figsize=(20,5))
scatter = weights_dataframe.plot.scatter(x="ecoli_cai_weights", y="bacillus_cai_weights", title="E.coli CAI weights vs. Bacillus CAI weights", ax=axes[0])
r, p = stats.spearmanr(weights_dataframe["ecoli_cai_weights"], weights_dataframe["bacillus_cai_weights"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = weights_dataframe.plot.scatter(x="ecoli_tai_weights", y="bacillus_tai_weights", title="E.coli tAI weights vs. Bacillus tAI weights", ax=axes[1])
# r, p = stats.spearmanr(weights_dataframe["ecoli_tai_weights"], weights_dataframe["bacillus_tai_weights"])
# scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')


merged_weights_codon_frequencies_df = pd.merge(weights_dataframe, ecoli_codon_frequencies, on='codon', how='inner')
merged_weights_codon_frequencies_df = pd.merge(merged_weights_codon_frequencies_df, bacillus_codon_frequencies, on='codon', how='inner', suffixes=("_ecoli", "_bacillus"))

fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))
scatter = merged_weights_codon_frequencies_df.plot.scatter(x="ecoli_cai_weights", y="frequency_ecoli", title="E.coli CAI weights vs. E.coli codon frequencies", ax=axes[0,0])
r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_cai_weights"], merged_weights_codon_frequencies_df["frequency_ecoli"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_weights_codon_frequencies_df.plot.scatter(x="bacillus_cai_weights", y="frequency_bacillus", title="Bacillus CAI weights vs. Bacillus codon frequencies", ax=axes[0,1])
r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_cai_weights"], merged_weights_codon_frequencies_df["frequency_bacillus"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = merged_weights_codon_frequencies_df.plot.scatter(x="ecoli_tai_weights", y="frequency_ecoli", title="E.coli tAI weights vs. E.coli codon frequencies", ax=axes[1,0])
# r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_tai_weights"], merged_weights_codon_frequencies_df["frequency_ecoli"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = merged_weights_codon_frequencies_df.plot.scatter(x="bacillus_tai_weights", y="frequency_bacillus", title="Bacillus tAI weights vs. Bacillus codon frequencies", ax=axes[1,1])
# r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_tai_weights"], merged_weights_codon_frequencies_df["frequency_bacillus"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')


merged_weights_codon_frequencies_df = pd.merge(weights_dataframe, ecoli_highly_expressed_genes_codon_frequencies, on='codon', how='inner')
merged_weights_codon_frequencies_df = pd.merge(merged_weights_codon_frequencies_df, bacillus_highly_expressed_genes_codon_frequencies, on='codon', how='inner', suffixes=("_ecoli", "_bacillus"))

fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))
scatter = merged_weights_codon_frequencies_df.plot.scatter(x="ecoli_cai_weights", y="frequency_ecoli", title="E.coli CAI weights vs. E.coli codon frequencies in highly expressed genes", ax=axes[0,0])
r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_cai_weights"], merged_weights_codon_frequencies_df["frequency_ecoli"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_weights_codon_frequencies_df.plot.scatter(x="bacillus_cai_weights", y="frequency_bacillus", title="Bacillus CAI weights vs. Bacillus codon frequencies in highly expressed genes", ax=axes[0,1])
r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_cai_weights"], merged_weights_codon_frequencies_df["frequency_bacillus"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = merged_weights_codon_frequencies_df.plot.scatter(x="ecoli_tai_weights", y="frequency_ecoli", title="E.coli tAI weights vs. E.coli codon frequencies in highly expressed genes", ax=axes[1,0])
# r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_tai_weights"], merged_weights_codon_frequencies_df["frequency_ecoli"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = merged_weights_codon_frequencies_df.plot.scatter(x="bacillus_tai_weights", y="frequency_bacillus", title="Bacillus tAI weights vs. Bacillus codon frequencies in highly expressed genes", ax=axes[1,1])
# r, p = stats.spearmanr(merged_weights_codon_frequencies_df["ecoli_tai_weights"], merged_weights_codon_frequencies_df["frequency_bacillus"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')


ecoli_cai_scores, ecoli_tai_scores = get_cub_scores(r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\Escherichia-coli_True_parsed.json")
bacillus_cai_scores, bacillus_tai_scores = get_cub_scores(r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\Bacillus-subtilis_False_parsed.json")


def get_scores(estimated_expression_dict, cai_scores_dict, tai_scores_dict, expression_type, selected_genes=None):
    all_cds = cai_scores_dict.keys()
    selected_genes = selected_genes or all_cds
    filtered_cds = []
    expression_levels = []
    
    expression_levels_key = "mrna_level" if expression_type == "mrna_levels" else "pa"
    for k in all_cds:
        if not any(k.startswith(x) for x in selected_genes):
            continue
        short_name = k.split(" ")[0]
        # print(f"k: {k}, sort_name: {short_name}")
        for gene_name, expression in estimated_expression_dict.items():
            if gene_name.startswith(short_name):
                expression_levels.append(expression[expression_levels_key])
                filtered_cds.append(k)
                break

        
    scores_expression_levels_df_dict = {
        "gene": filtered_cds, 
        "cai_scores": [cai_scores_dict[k] for k in filtered_cds],
        # "tai_scores": [tai_scores_dict[k] for k in filtered_cds],
        expression_levels_key: expression_levels,
    }
    return pd.DataFrame(scores_expression_levels_df_dict)


expression_type = "mrna_levels"
ecoli_estimated_expression = get_estimated_expression(endogenous_genes_fasta_file_path=ecoli_endogenous_genes_fasta_file_path,
                                                      expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\ecoli_mrna_level.csv",
                                                      expression_type=expression_type)
ecoli_scores_dataframe =  get_scores(estimated_expression_dict=ecoli_estimated_expression, cai_scores_dict=ecoli_cai_scores, tai_scores_dict=ecoli_tai_scores, expression_type=expression_type)
print(f"Ecoli mrna levels stats:\n records:{len(ecoli_scores_dataframe['mrna_level'])}\n min: {ecoli_scores_dataframe['mrna_level'].min()} \n max: {ecoli_scores_dataframe['mrna_level'].max()}\n mean: {ecoli_scores_dataframe['mrna_level'].mean()}\n std: {ecoli_scores_dataframe['mrna_level'].std()}")  
ecoli_scores_dataframe["mrna_level"].sort_values(ascending=False).reset_index(drop=True).plot()

bacillus_estimated_expression = get_estimated_expression(endogenous_genes_fasta_file_path=bacillus_endogenous_genes_fasta_file_path,
                                                         expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\bacillus_mrna_level.csv",
                                                          expression_type=expression_type)
bacillus_scores_dataframe =  get_scores(estimated_expression_dict=bacillus_estimated_expression, cai_scores_dict=bacillus_cai_scores, tai_scores_dict=bacillus_tai_scores, expression_type=expression_type)
print(f"Bacillus mrna levels stats:\n records:{len(bacillus_scores_dataframe['mrna_level'])}\n min: {bacillus_scores_dataframe['mrna_level'].min()} \n max: {bacillus_scores_dataframe['mrna_level'].max()}\n mean: {bacillus_scores_dataframe['mrna_level'].mean()}\n std: {bacillus_scores_dataframe['mrna_level'].std()}")  
bacillus_scores_dataframe["mrna_level"].sort_values(ascending=False).reset_index(drop=True).plot()


fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))

ecoli_scores_dataframe["mrna_level_logarithmic_base10"] = np.log10(ecoli_scores_dataframe['mrna_level'])
bacillus_scores_dataframe["mrna_level_logarithmic_base10"] = np.log10(bacillus_scores_dataframe['mrna_level'])

scatter = ecoli_scores_dataframe.plot.scatter(x="cai_scores", y="mrna_level_logarithmic_base10", title="E.coli mRNA levels vs. E.coli CAI scores", ax=axes[0,0])
r, p = stats.spearmanr(ecoli_scores_dataframe["cai_scores"], ecoli_scores_dataframe["mrna_level"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = ecoli_scores_dataframe.plot.scatter(x="tai_scores", y="mrna_level", title="E.coli mRNA levels vs. E.coli tAI scores", ax=axes[0,1])
# r, p = stats.spearmanr(ecoli_scores_dataframe["tai_scores"], ecoli_scores_dataframe["mrna_level"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = bacillus_scores_dataframe.plot.scatter(x="cai_scores", y="mrna_level_logarithmic_base10", title="Baciilus mRNA levels vs. Baciilus CAI scores", ax=axes[1,0])
r, p = stats.spearmanr(bacillus_scores_dataframe["cai_scores"], bacillus_scores_dataframe["mrna_level"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = bacillus_scores_dataframe.plot.scatter(x="tai_scores", y="mrna_level", title="Baciilus mRNA levels vs. Baciilus tAI scores", ax=axes[1,1])
# r, p = stats.spearmanr(bacillus_scores_dataframe["tai_scores"], bacillus_scores_dataframe["mrna_level"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')


expression_type = "mrna_levels"
ecoli_estimated_expression = get_estimated_expression(endogenous_genes_fasta_file_path=ecoli_endogenous_genes_fasta_file_path,
                                                      expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\ecoli_mrna_level.csv",
                                                      expression_type=expression_type)
ecoli_highly_expressed_scores_dataframe =  get_scores(estimated_expression_dict=ecoli_estimated_expression, cai_scores_dict=ecoli_cai_scores, tai_scores_dict=ecoli_tai_scores, selected_genes=ecoli_highly_expressed_genes.keys(), expression_type=expression_type)
print(f"Ecoli mrna levels stats:\n records:{len(ecoli_highly_expressed_scores_dataframe['mrna_level'])}\n min: {ecoli_highly_expressed_scores_dataframe['mrna_level'].min()} \n max: {ecoli_highly_expressed_scores_dataframe['mrna_level'].max()}\n mean: {ecoli_highly_expressed_scores_dataframe['mrna_level'].mean()}\n std: {ecoli_highly_expressed_scores_dataframe['mrna_level'].std()}")  
ecoli_highly_expressed_scores_dataframe["mrna_level"].sort_values(ascending=False).reset_index(drop=True).plot()
bacillus_estimated_expression = get_estimated_expression(endogenous_genes_fasta_file_path=bacillus_endogenous_genes_fasta_file_path,
                                                      expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\bacillus_mrna_level.csv",
                                                        expression_type=expression_type)
bacillus_highly_expressed_scores_dataframe =  get_scores(estimated_expression_dict=bacillus_estimated_expression, cai_scores_dict=bacillus_cai_scores, tai_scores_dict=bacillus_tai_scores, selected_genes=bacillus_highly_expressed_genes.keys(), expression_type=expression_type)
print(f"Bacillus mrna levels stats:\n records:{len(bacillus_highly_expressed_scores_dataframe['mrna_level'])}\n min: {bacillus_highly_expressed_scores_dataframe['mrna_level'].min()} \n max: {bacillus_highly_expressed_scores_dataframe['mrna_level'].max()}\n mean: {bacillus_highly_expressed_scores_dataframe['mrna_level'].mean()}\n std: {bacillus_highly_expressed_scores_dataframe['mrna_level'].std()}")  
bacillus_highly_expressed_scores_dataframe["mrna_level"].sort_values(ascending=False).reset_index(drop=True).plot()


fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))

ecoli_highly_expressed_scores_dataframe["mrna_level_logarithmic_base10"] = np.log10(ecoli_highly_expressed_scores_dataframe['mrna_level'])
bacillus_highly_expressed_scores_dataframe["mrna_level_logarithmic_base10"] = np.log10(bacillus_highly_expressed_scores_dataframe['mrna_level'])

scatter = ecoli_highly_expressed_scores_dataframe.plot.scatter(x="cai_scores", y="mrna_level_logarithmic_base10", title="E.coli mRNA levels (Highly expressed) vs. E.coli CAI scores", ax=axes[0,0])
r, p = stats.spearmanr(ecoli_highly_expressed_scores_dataframe["cai_scores"], ecoli_highly_expressed_scores_dataframe["mrna_level"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = ecoli_highly_expressed_scores_dataframe.plot.scatter(x="tai_scores", y="mrna_level", title="E.coli mRNA levels (Highly expressed) vs. E.coli tAI scores", ax=axes[0,1])
# r, p = stats.spearmanr(ecoli_highly_expressed_scores_dataframe["tai_scores"], ecoli_highly_expressed_scores_dataframe["mrna_level"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = bacillus_highly_expressed_scores_dataframe.plot.scatter(x="cai_scores", y="mrna_level_logarithmic_base10", title="Baciilus mRNA levels (Highly expressed) vs. Baciilus CAI scores", ax=axes[1,0])
r, p = stats.spearmanr(bacillus_highly_expressed_scores_dataframe["cai_scores"], bacillus_highly_expressed_scores_dataframe["mrna_level"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = bacillus_highly_expressed_scores_dataframe.plot.scatter(x="tai_scores", y="mrna_level", title="Baciilus mRNA levels (Highly expressed) vs. Baciilus tAI scores", ax=axes[1,1])
# r, p = stats.spearmanr(bacillus_highly_expressed_scores_dataframe["tai_scores"], bacillus_highly_expressed_scores_dataframe["mrna_level"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')


expression_type = "protein_abundance"
ecoli_estimated_expression_pa = get_estimated_expression(endogenous_genes_fasta_file_path=ecoli_endogenous_genes_fasta_file_path,
                                                         expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\PA_ecoli.json",
                                                         expression_type=expression_type)
ecoli_scores_dataframe_pa =  get_scores(estimated_expression_dict=ecoli_estimated_expression_pa, cai_scores_dict=ecoli_cai_scores, tai_scores_dict=ecoli_tai_scores, expression_type=expression_type)
print(f"Ecoli protein abundance stats:\n records:{len(ecoli_scores_dataframe_pa['pa'])}\n min: {ecoli_scores_dataframe_pa['pa'].min()} \n max: {ecoli_scores_dataframe_pa['pa'].max()}\n mean: {ecoli_scores_dataframe_pa['pa'].mean()}\n std: {ecoli_scores_dataframe_pa['pa'].std()}")  
ecoli_scores_dataframe_pa["pa"].sort_values(ascending=False).reset_index(drop=True).plot()

bacillus_estimated_expression_pa = get_estimated_expression(endogenous_genes_fasta_file_path=bacillus_endogenous_genes_fasta_file_path,
                                                         expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\PA_bacillus.json",
                                                          expression_type=expression_type)
bacillus_scores_dataframe_pa =  get_scores(estimated_expression_dict=bacillus_estimated_expression_pa, cai_scores_dict=bacillus_cai_scores, tai_scores_dict=bacillus_tai_scores, expression_type=expression_type)
print(f"Bacillus protein abundance stats:\n records:{len(bacillus_scores_dataframe_pa['pa'])}\n min: {bacillus_scores_dataframe_pa['pa'].min()} \n max: {bacillus_scores_dataframe_pa['pa'].max()}\n mean: {bacillus_scores_dataframe_pa['pa'].mean()}\n std: {bacillus_scores_dataframe_pa['pa'].std()}")  
bacillus_scores_dataframe_pa["pa"].sort_values(ascending=False).reset_index(drop=True).plot()


fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))

ecoli_scores_dataframe_pa["pa_logarithmic_base10"] = np.log10(ecoli_scores_dataframe_pa['pa'])
bacillus_scores_dataframe_pa["pa_logarithmic_base10"] = np.log10(bacillus_scores_dataframe_pa['pa'])
                                                              

scatter = ecoli_scores_dataframe_pa.plot.scatter(x="cai_scores", y="pa_logarithmic_base10", title="E.coli protein abundance vs. E.coli CAI scores", ax=axes[0,0])
r, p = stats.spearmanr(ecoli_scores_dataframe_pa["cai_scores"], ecoli_scores_dataframe_pa["pa"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = ecoli_scores_dataframe_pa.plot.scatter(x="tai_scores", y="pa_logarithmic_base10", title="E.coli protein abundance vs. E.coli tAI scores", ax=axes[0,1])
# r, p = stats.spearmanr(ecoli_scores_dataframe_pa["tai_scores"], ecoli_scores_dataframe_pa["pa"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = bacillus_scores_dataframe_pa.plot.scatter(x="cai_scores", y="pa_logarithmic_base10", title="Baciilus protein abundance vs. Baciilus CAI scores", ax=axes[1,0])
r, p = stats.spearmanr(bacillus_scores_dataframe_pa["cai_scores"], bacillus_scores_dataframe_pa["pa"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = bacillus_scores_dataframe_pa.plot.scatter(x="tai_scores", y="pa_logarithmic_base10", title="Baciilus protein abundance vs. Baciilus tAI scores", ax=axes[1,1])
# r, p = stats.spearmanr(bacillus_scores_dataframe_pa["tai_scores"], bacillus_scores_dataframe_pa["pa"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')


expression_type = "protein_abundance"
ecoli_estimated_expression_pa = get_estimated_expression(endogenous_genes_fasta_file_path=ecoli_endogenous_genes_fasta_file_path,
                                                         expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\PA_ecoli.json",
                                                         expression_type=expression_type)
ecoli_scores_dataframe_pa_highly_expressed =  get_scores(estimated_expression_dict=ecoli_estimated_expression_pa, cai_scores_dict=ecoli_cai_scores, tai_scores_dict=ecoli_tai_scores, expression_type=expression_type, selected_genes=ecoli_highly_expressed_genes.keys())
print(f"Ecoli protein abundance stats:\n records:{len(ecoli_scores_dataframe_pa_highly_expressed['pa'])}\n min: {ecoli_scores_dataframe_pa_highly_expressed['pa'].min()} \n max: {ecoli_scores_dataframe_pa_highly_expressed['pa'].max()}\n mean: {ecoli_scores_dataframe_pa_highly_expressed['pa'].mean()}\n std: {ecoli_scores_dataframe_pa_highly_expressed['pa'].std()}")  
ecoli_scores_dataframe_pa_highly_expressed["pa"].sort_values(ascending=False).reset_index(drop=True).plot()

bacillus_estimated_expression_pa = get_estimated_expression(endogenous_genes_fasta_file_path=bacillus_endogenous_genes_fasta_file_path,
                                                         expression_levels_file_path=r"C:\projects\Igem_TAU_2021_moran\analysis\example_data\PA_bacillus.json",
                                                          expression_type=expression_type)
bacillus_scores_dataframe_pa_highly_expressed =  get_scores(estimated_expression_dict=bacillus_estimated_expression_pa, cai_scores_dict=bacillus_cai_scores, tai_scores_dict=bacillus_tai_scores, expression_type=expression_type, selected_genes=bacillus_highly_expressed_genes.keys())
print(f"Bacillus protein abundance stats:\n records:{len(bacillus_scores_dataframe_pa_highly_expressed['pa'])}\n min: {bacillus_scores_dataframe_pa_highly_expressed['pa'].min()} \n max: {bacillus_scores_dataframe_pa_highly_expressed['pa'].max()}\n mean: {bacillus_scores_dataframe_pa_highly_expressed['pa'].mean()}\n std: {bacillus_scores_dataframe_pa_highly_expressed['pa'].std()}")  
bacillus_scores_dataframe_pa_highly_expressed["pa"].sort_values(ascending=False).reset_index(drop=True).plot()


fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))

ecoli_scores_dataframe_pa_highly_expressed["pa_logarithmic_base10"] = np.log10(ecoli_scores_dataframe_pa_highly_expressed['pa'])
bacillus_scores_dataframe_pa_highly_expressed["pa_logarithmic_base10"] = np.log10(bacillus_scores_dataframe_pa_highly_expressed['pa'])

scatter = ecoli_scores_dataframe_pa_highly_expressed.plot.scatter(x="cai_scores", y="pa_logarithmic_base10", title="E.coli protein abundance vs. E.coli CAI scores - Highly expressed", ax=axes[0,0])
r, p = stats.spearmanr(ecoli_scores_dataframe_pa_highly_expressed["cai_scores"], ecoli_scores_dataframe_pa_highly_expressed["pa"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = ecoli_scores_dataframe_pa_highly_expressed.plot.scatter(x="tai_scores", y="pa_logarithmic_base10", title="E.coli protein abundance vs. E.coli tAI scores - Highly expressed", ax=axes[0,1])
# r, p = stats.spearmanr(ecoli_scores_dataframe_pa_highly_expressed["tai_scores"], ecoli_scores_dataframe_pa_highly_expressed["pa"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = bacillus_scores_dataframe_pa_highly_expressed.plot.scatter(x="cai_scores", y="pa_logarithmic_base10", title="Baciilus protein abundance vs. Baciilus CAI scores - Highly expressed ", ax=axes[1,0])
r, p = stats.spearmanr(bacillus_scores_dataframe_pa_highly_expressed["cai_scores"], bacillus_scores_dataframe_pa_highly_expressed["pa"])
scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')

# scatter = bacillus_scores_dataframe_pa.plot.scatter(x="tai_scores", y="pa_logarithmic_base10", title="Baciilus protein abundance vs. Baciilus tAI scores - Highly expressed", ax=axes[1,1])
# r, p = stats.spearmanr(bacillus_scores_dataframe_pa_highly_expressed["tai_scores"], bacillus_scores_dataframe_pa_highly_expressed["pa"])
# scatter.annotate('r = {:.4f}, p={}'.format(r, p), xy=(0.7, 0.9), xycoords='axes fraction')








# Reference aa sequence is taken from: https://www.addgene.org/browse/sequence_vdb/6496/
reference_mcherry_protein_seq  = "MVSKGEEDNMAIIKEFMRFKVHMEGSVNGHEFEIEGEGEGRPYEGTQTAKLKVTKGGPLPFAWDILSPQFMYGSKAYVKHPADIPDYLKLSFPEGFKWERVMNFEDGGVVTVTQDSSLQDGEFIYKVKLRGTNFPSDGPVMQKKTMGWEASSERMYPEDGALKGEIKQRLKLKDGGHYDAEVKTTYKAKKPVQLPGAYNVNIKLDITSHNEDYTIVEQYERAEGRHSTGGMDELYK*"
formatted_reference_mcherry_protein_seq = reference_mcherry_protein_seq.replace("*", "_")

for i in range(1, len(formatted_reference_mcherry_protein_seq)):
    prev_aa = formatted_reference_mcherry_protein_seq[i-1]
    aa = formatted_reference_mcherry_protein_seq[i]
    if prev_aa == aa:
        print(f"aa {aa} is the same as it previous, location {i}")

translated_original_mcherry_seq = shared_functions_and_vars.translate(original_mcherry_seq)
     
assert translated_original_mcherry_seq == formatted_reference_mcherry_protein_seq





with open(mcherry_variants_file, "r") as mcherry_variants_fasta_file:
    mcherry_variants_dict = SeqIO.to_dict(SeqIO.parse(mcherry_variants_fasta_file, "fasta"))

for variant_name, variant_seq in mcherry_variants_dict.items():
    translated_seq = shared_functions_and_vars.translate(variant_seq.seq)
    if translated_seq != formatted_reference_mcherry_protein_seq:
        raise ValueError(f"Variant {variant_name} is not coding to correct sequence. \nReference sequence: {formatted_reference_mcherry_protein_seq} \nVariant sequence: {translated_seq}")





bacillus_codon_frequencies
ecoli_codon_frequencies

# import codonbias as cb
# codon_counter = cb.scores.CodonCounter(genetic_code=1, ignore_stop=False)
# codon_counter.count(ref_seq)\
#             .get_aa_table(normed=True, pseudocount=pseudocount)\
#             .groupby('aa')


# # Skip if not taking the original sequence prefix
# skipped_codons = 12
# skipped_codons_in_nt = 12 * 3
# for variant_name, variant_seq in mcherry_variants_dict.items():
#     assert original_mcherry_seq[:skipped_codons_in_nt].lower() == str(variant_seq.seq)[:skipped_codons_in_nt].lower(), f"variant {variant_name} has different initiation prefix"
#     assert original_mcherry_seq.lower() != str(variant_seq.seq).lower(), "variant is identical to the original sequence"
#     # print(original_mcherry_seq[:skipped_codons_in_nt].lower())
#     # print(str(variant_seq.seq)[:skipped_codons_in_nt].lower())








original_mcherry_codon_frequencies = calculate_codon_frequencies((str(original_mcherry_seq),))
original_mcherry_codon_frequencies = original_mcherry_codon_frequencies.sort_values("codon")

original_mcherry_codon_frequencies_lamda = calculate_codon_frequencies_lamda((str(original_mcherry_seq),))
original_mcherry_codon_frequencies_lamda = original_mcherry_codon_frequencies_lamda.sort_values("codon")


original_bacillus_dist = calculate_distance_between_frequencies(original_mcherry_codon_frequencies["frequency"].reset_index(drop=True), bacillus_codon_frequencies["frequency"].reset_index(drop=True))
original_ecoli_dist = calculate_distance_between_frequencies(original_mcherry_codon_frequencies["frequency"].reset_index(drop=True), ecoli_codon_frequencies["frequency"].reset_index(drop=True))

original_bacillus_dist_lamda = calculate_distance_between_frequencies(original_mcherry_codon_frequencies_lamda["frequency"].reset_index(drop=True), bacillus_codon_frequencies_lamda["frequency"].reset_index(drop=True))
original_ecoli_dist_lamda = calculate_distance_between_frequencies(original_mcherry_codon_frequencies_lamda["frequency"].reset_index(drop=True), ecoli_codon_frequencies_lamda["frequency"].reset_index(drop=True))

original_bacillus_dist_highly_expressed = calculate_distance_between_frequencies(original_mcherry_codon_frequencies["frequency"].reset_index(drop=True), bacillus_highly_expressed_genes_codon_frequencies["frequency"].reset_index(drop=True))
original_ecoli_dist_highly_expressed = calculate_distance_between_frequencies(original_mcherry_codon_frequencies["frequency"].reset_index(drop=True), ecoli_highly_expressed_genes_codon_frequencies["frequency"].reset_index(drop=True))

original_bacillus_dist_highly_expressed_lamda = calculate_distance_between_frequencies(original_mcherry_codon_frequencies_lamda["frequency"].reset_index(drop=True), bacillus_highly_expressed_genes_codon_frequencies_lamda["frequency"].reset_index(drop=True))
original_ecoli_dist_highly_expressed_lamda = calculate_distance_between_frequencies(original_mcherry_codon_frequencies_lamda["frequency"].reset_index(drop=True), ecoli_highly_expressed_genes_codon_frequencies_lamda["frequency"].reset_index(drop=True))


df_dict = get_variants_df_dict(mcherry_variants_file, ecoli_codon_frequencies, bacillus_codon_frequencies, ecoli_codon_frequencies_lamda, bacillus_codon_frequencies_lamda)

ecoli_opt_df_dict = {key: value for key, value in df_dict.items() if not key.endswith("for_optimized_bacillus")}
ecoli_opt_variants_dataframe = pd.DataFrame(ecoli_opt_df_dict)

bacillus_opt_df_dict = {key: value for key, value in df_dict.items() if not key.endswith("for_optimized_ecoli")}
bacillus_opt_variants_dataframe = pd.DataFrame(bacillus_opt_df_dict)

df_dict_highly_expressed = get_variants_df_dict(mcherry_variants_file, ecoli_highly_expressed_genes_codon_frequencies, bacillus_highly_expressed_genes_codon_frequencies, ecoli_highly_expressed_genes_codon_frequencies_lamda, bacillus_highly_expressed_genes_codon_frequencies_lamda)

ecoli_opt_df_dict_highly_expressed = {key: value for key, value in df_dict_highly_expressed.items() if not key.endswith("for_optimized_bacillus")}
ecoli_opt_variants_dataframe_highly_expressed = pd.DataFrame(ecoli_opt_df_dict_highly_expressed)

bacillus_opt_df_dict_highly_expressed = {key: value for key, value in df_dict_highly_expressed.items() if not key.endswith("for_optimized_ecoli")}
bacillus_opt_variants_dataframe_highly_expressed = pd.DataFrame(bacillus_opt_df_dict_highly_expressed)


bacillus_A = bacillus_codon_frequencies[bacillus_codon_frequencies["amino_acid"] == "A"].sort_values("codon")
x = pd.DataFrame(get_codon_frequencies_df_dict(mcherry_variants_file))
x_A = x[x["amino_acid"] == "A"].sort_values("codon")
original_A = original_mcherry_codon_frequencies[original_mcherry_codon_frequencies["amino_acid"] == "A"].sort_values("codon")


original_dist = calculate_distance_between_frequencies(original_A["frequency"].reset_index(drop=True), bacillus_A["frequency"].reset_index(drop=True))
variant_dist = calculate_distance_between_frequencies(x_A["frequency"].reset_index(drop=True), bacillus_A["frequency"].reset_index(drop=True))
bacillus_A


original_A


x_A


print(f"original seq dist: {original_dist}, variant dist: {variant_dist}")





df_dict = get_cub_score_df_dict(mcherry_root_dir)
cub_ecoli_opt_df_dict = {key.replace("_for_optimized_ecoli", ""): value for key, value in df_dict.items() if not (key.endswith("_for_optimized_bacillus") or key.startswith("evaluation_score"))}
cub_ecoli_opt_variants_dataframe = pd.DataFrame(cub_ecoli_opt_df_dict)

evaluation_ecoli_opt_df_dict = {key: value for key, value in df_dict.items() if key in ["variant", "average_distance_evaluation_score_for_optimized_ecoli", "ratio_evaluation_score_for_optimized_ecoli"]}
evaluation_ecoli_opt_variants_dataframe = pd.DataFrame(evaluation_ecoli_opt_df_dict)

cub_bacillus_opt_df_dict = {key.replace("_for_optimized_bacillus", ""): value for key, value in df_dict.items() if not (key.endswith("for_optimized_ecoli") or key.startswith("evaluation_score"))}
cub_bacillus_opt_variants_dataframe = pd.DataFrame(cub_bacillus_opt_df_dict)

evaluation_bacillus_opt_df_dict = {key: value for key, value in df_dict.items() if key in ["variant", "average_distance_evaluation_score_for_optimized_bacillus", "ratio_evaluation_score_for_optimized_bacillus"]}
evaluation_bacillus_opt_variants_dataframe = pd.DataFrame(evaluation_bacillus_opt_df_dict)


evaluation_ecoli_opt_variants_dataframe





greater_alpha_df_dict = get_variants_df_dict(mcherry_greater_alpha_variants_file, ecoli_codon_frequencies, bacillus_codon_frequencies, ecoli_codon_frequencies_lamda, bacillus_codon_frequencies_lamda)
    
greater_alpha_ecoli_opt_df_dict = {key: value for key, value in greater_alpha_df_dict.items() if not key.endswith("for_optimized_bacillus")}
greater_alpha_ecoli_opt_variants_dataframe = pd.DataFrame(greater_alpha_ecoli_opt_df_dict)

greater_alpha_bacillus_opt_df_dict = {key: value for key, value in greater_alpha_df_dict.items() if not key.endswith("for_optimized_ecoli")}
greater_alpha_bacillus_opt_variants_dataframe = pd.DataFrame(greater_alpha_bacillus_opt_df_dict)

greater_alpha_df_dict_highly_expressed = get_variants_df_dict(mcherry_greater_alpha_variants_file, ecoli_highly_expressed_genes_codon_frequencies, bacillus_highly_expressed_genes_codon_frequencies, ecoli_highly_expressed_genes_codon_frequencies_lamda, bacillus_highly_expressed_genes_codon_frequencies_lamda)

greater_alpha_ecoli_opt_df_dict_highly_expressed = {key: value for key, value in greater_alpha_df_dict_highly_expressed.items() if not key.endswith("for_optimized_bacillus")}
greater_alpha_ecoli_opt_variants_dataframe_highly_expressed = pd.DataFrame(greater_alpha_ecoli_opt_df_dict_highly_expressed)

greater_alpha_bacillus_opt_df_dict_highly_expressed = {key: value for key, value in greater_alpha_df_dict_highly_expressed.items() if not key.endswith("for_optimized_ecoli")}
greater_alpha_bacillus_opt_variants_dataframe_highly_expressed = pd.DataFrame(greater_alpha_bacillus_opt_df_dict_highly_expressed)





greater_alpha_df_dict = get_cub_score_df_dict(mcherry_greater_alpha_root_dir)
cub_ecoli_opt_greater_alpha_df_dict = {key.replace("_for_optimized_ecoli", ""): value for key, value in greater_alpha_df_dict.items() if not (key.endswith("_for_optimized_bacillus") or key.startswith("evaluation_score"))}
cub_ecoli_opt_greater_alpha_variants_dataframe = pd.DataFrame(cub_ecoli_opt_greater_alpha_df_dict)

evaluation_ecoli_opt_greate_alpha_df_dict = {key: value for key, value in greater_alpha_df_dict.items() if key in ["variant", "average_distance_evaluation_score_for_optimized_ecoli", "ratio_evaluation_score_for_optimized_ecoli"]}
evaluation_ecoli_opt_greater_alpha_variants_dataframe = pd.DataFrame(evaluation_ecoli_opt_greate_alpha_df_dict)

cub_bacillus_opt_greater_alpha_df_dict = {key.replace("_for_optimized_bacillus", ""): value for key, value in greater_alpha_df_dict.items() if not (key.endswith("for_optimized_ecoli") or key.startswith("evaluation_score"))}
cub_bacillus_opt_greater_alpha_variants_dataframe = pd.DataFrame(cub_bacillus_opt_greater_alpha_df_dict)

evaluation_bacillus_opt_greater_alpha_df_dict = {key: value for key, value in greater_alpha_df_dict.items() if key in ["variant", "evaluation_score_for_optimized_bacillus", "average_distance_evaluation_score_for_optimized_bacillus", "ratio_evaluation_score_for_optimized_bacillus"]}
evaluation_bacillus_opt_greater_alpha_variants_dataframe = pd.DataFrame(evaluation_bacillus_opt_greater_alpha_df_dict)





lower_alpha_df_dict = get_variants_df_dict(mcherry_lower_alpha_variants_file,  ecoli_codon_frequencies, bacillus_codon_frequencies, ecoli_codon_frequencies_lamda, bacillus_codon_frequencies_lamda)
    
lower_alpha_ecoli_opt_df_dict = {key: value for key, value in lower_alpha_df_dict.items() if not key.endswith("for_optimized_bacillus")}
lower_alpha_ecoli_opt_variants_dataframe = pd.DataFrame(lower_alpha_ecoli_opt_df_dict)

lower_alpha_bacillus_opt_df_dict = {key: value for key, value in lower_alpha_df_dict.items() if not key.endswith("for_optimized_ecoli")}
lower_alpha_bacillus_opt_variants_dataframe = pd.DataFrame(lower_alpha_bacillus_opt_df_dict)

lower_alpha_df_dict_highly_expressed = get_variants_df_dict(mcherry_lower_alpha_variants_file, ecoli_highly_expressed_genes_codon_frequencies, bacillus_highly_expressed_genes_codon_frequencies, ecoli_highly_expressed_genes_codon_frequencies_lamda, bacillus_highly_expressed_genes_codon_frequencies_lamda)

lower_alpha_ecoli_opt_df_dict_highly_expressed = {key: value for key, value in lower_alpha_df_dict_highly_expressed.items() if not key.endswith("for_optimized_bacillus")}
lower_alpha_ecoli_opt_variants_dataframe_highly_expressed = pd.DataFrame(lower_alpha_ecoli_opt_df_dict_highly_expressed)

lower_alpha_bacillus_opt_df_dict_highly_expressed = {key: value for key, value in lower_alpha_df_dict_highly_expressed.items() if not key.endswith("for_optimized_ecoli")}
lower_alpha_bacillus_opt_variants_dataframe_highly_expressed = pd.DataFrame(lower_alpha_bacillus_opt_df_dict_highly_expressed)


lower_alpha_df_dict = get_cub_score_df_dict(mcherry_lower_alpha_root_dir)
cub_ecoli_opt_lower_alpha_df_dict = {key.replace("_for_optimized_ecoli", ""): value for key, value in lower_alpha_df_dict.items() if not (key.endswith("_for_optimized_bacillus") or key.startswith("evaluation_score"))}
cub_ecoli_opt_lower_alpha_variants_dataframe = pd.DataFrame(cub_ecoli_opt_lower_alpha_df_dict)

evaluation_ecoli_opt_lower_alpha_df_dict = {key: value for key, value in lower_alpha_df_dict.items() if key in ["variant", "average_distance_evaluation_score_for_optimized_ecoli", "ratio_evaluation_score_for_optimized_ecoli"]}
evaluation_ecoli_opt_lower_alpha_variants_dataframe = pd.DataFrame(evaluation_ecoli_opt_lower_alpha_df_dict)

cub_bacillus_opt_lower_alpha_df_dict = {key.replace("_for_optimized_bacillus", ""): value for key, value in lower_alpha_df_dict.items() if not (key.endswith("for_optimized_ecoli") or key.startswith("evaluation_score"))}
cub_bacillus_opt_lower_alpha_variants_dataframe = pd.DataFrame(cub_bacillus_opt_lower_alpha_df_dict)

evaluation_bacillus_opt_lower_alpha_df_dict = {key: value for key, value in lower_alpha_df_dict.items() if key in ["variant", "average_distance_evaluation_score_for_optimized_bacillus", "ratio_evaluation_score_for_optimized_bacillus"]}
evaluation_bacillus_opt_lower_alpha_variants_dataframe = pd.DataFrame(evaluation_bacillus_opt_lower_alpha_df_dict)





fig, axes = plt.subplots(nrows=1,ncols=2, figsize=(27,10))

fig.suptitle("Variants Evaluation Scores")
evaluation_ecoli_opt_variants_dataframe.plot("variant", kind="bar", ax=axes[0], title="E.coli optimized")
evaluation_bacillus_opt_variants_dataframe.plot("variant", kind="bar", ax=axes[1], title="Bacillus optimized")
axes[0].set_ylim([0, 4.5])
axes[1].set_ylim([0, 4.5])
# evaluation_bacillus_opt_variants_dataframe.plot("variant", kind="barh", ax=axes[0])


# evaluation_ecoli_opt_variants_dataframe.plot("variant", kind="barh", ax=axes[0], title="E.coli optimized")
# evaluation_bacillus_opt_variants_dataframe.plot("variant", kind="barh", ax=axes[1], title="Bacillus optimized")

# evaluation_ecoli_opt_variants_dataframe.plot("variant", kind="barh", ax=axes[0], title="E.coli optimized")
# evaluation_bacillus_opt_variants_dataframe.plot("variant", kind="barh", ax=axes[1], title="Bacillus optimized", color="#dc4d01")





random_sequences_num = 1000
random_sequences = [shared_functions_and_vars.synonymous_codon_permutation(original_mcherry_seq) for _ in range(random_sequences_num)]

random_evaluation_scores = {}
dummy_optimization_method = "single_codon_ratio"
dummy_output_path = "results/mcherry_debug"
for optimization_cub_index in [
    "CAI", 
    # "tAI",
]:
    for direction in [True, False]:
        user_input = get_user_input(
            optimization_method=dummy_optimization_method,
            optimization_cub_index=optimization_cub_index,
            is_ecoli_optimized=direction,
            orf_sequence=str(original_mcherry_seq),
            output_path=dummy_output_path,
            should_use_protein_abundance=True,
        )
        key = f"{optimization_cub_index}_ecoli_opt_{direction}"
        random_evaluation_scores[key] = []
        for random_sequence in random_sequences:
            evaluation_score = evaluate_score_for_sequence(
                sequence=random_sequence,
                user_input=user_input,
                optimization_method=dummy_optimization_method,
                optimization_cub_index=optimization_cub_index,
                is_ecoli_optimized=direction,
                orf_sequence=original_mcherry_seq,
                output_path=dummy_output_path,
            )
            random_evaluation_scores[key].append(evaluation_score)


def get_random_scores_df(is_ecoli_opt: bool):
    random_scores = defaultdict(dict)
    variants = evaluation_ecoli_opt_variants_dataframe["variant"]
    for key, values in random_evaluation_scores.items():
        if str(is_ecoli_opt) not in key:
            continue
        average_distance_scores = [x.average_distance_score for x  in values]
        average_distance_average = np.average(average_distance_scores)
        average_distance_std =  np.std(average_distance_scores)
        ratio_scores = [x.ratio_score for x  in values]
        ratio_average = np.average(ratio_scores)
        ratio_std =  np.std(ratio_scores)
        for variant in variants:
            if variant.startswith(key.split("_")[0]):
                random_scores[variant]["average_distance_average"] = average_distance_average
                random_scores[variant]["average_distance_std"] = average_distance_std
                random_scores[variant]["ratio_average"] = ratio_average
                random_scores[variant]["ratio_std"] = ratio_std
    
    random_scores_df_dict = {
        "variant": variants, 
        "random_average_distance_average": [random_scores[variant]["average_distance_average"] for variant in variants],
        "random_average_distance_std": [random_scores[variant]["average_distance_std"] for variant in variants],
        "random_ratio_average": [random_scores[variant]["ratio_average"] for variant in variants],
        "random_ratio_std": [random_scores[variant]["ratio_std"] for variant in variants],
    }

    return pd.DataFrame(random_scores_df_dict)   

ecoli_opt_random_scores_dataframe = get_random_scores_df(is_ecoli_opt=True)
new_evaluation_ecoli_opt_variants_dataframe = pd.merge(evaluation_ecoli_opt_variants_dataframe, ecoli_opt_random_scores_dataframe, on='variant', how='inner')
new_evaluation_ecoli_opt_variants_dataframe["average_distance_zscore"] = (new_evaluation_ecoli_opt_variants_dataframe["average_distance_evaluation_score_for_optimized_ecoli"] - new_evaluation_ecoli_opt_variants_dataframe["random_average_distance_average"]) / new_evaluation_ecoli_opt_variants_dataframe["random_average_distance_std"]
new_evaluation_ecoli_opt_variants_dataframe["ratio_zscore"] = (new_evaluation_ecoli_opt_variants_dataframe["ratio_evaluation_score_for_optimized_ecoli"] - new_evaluation_ecoli_opt_variants_dataframe["random_ratio_average"]) / new_evaluation_ecoli_opt_variants_dataframe["random_ratio_std"]


bacillus_opt_random_scores_dataframe = get_random_scores_df(is_ecoli_opt=False)
new_evaluation_bacillus_opt_variants_dataframe = pd.merge(evaluation_bacillus_opt_variants_dataframe, bacillus_opt_random_scores_dataframe, on='variant', how='inner')
new_evaluation_bacillus_opt_variants_dataframe["average_distance_zscore"] = (new_evaluation_bacillus_opt_variants_dataframe["average_distance_evaluation_score_for_optimized_bacillus"] - new_evaluation_bacillus_opt_variants_dataframe["random_average_distance_average"]) / new_evaluation_bacillus_opt_variants_dataframe["random_average_distance_std"]
new_evaluation_bacillus_opt_variants_dataframe["ratio_zscore"] = (new_evaluation_bacillus_opt_variants_dataframe["ratio_evaluation_score_for_optimized_bacillus"] - new_evaluation_bacillus_opt_variants_dataframe["random_ratio_average"]) / new_evaluation_bacillus_opt_variants_dataframe["random_ratio_std"]




new_evaluation_ecoli_opt_variants_dataframe





new_evaluation_bacillus_opt_variants_dataframe


for key, values in random_evaluation_scores.items():
    variants = evaluation_ecoli_opt_variants_dataframe["variant"]
    for score_type in ["average_distance", "ratio"]:
        scores = [getattr(x, f"{score_type}_score") for x  in values]
        for variant in variants:
            if variant.startswith(key.split("_")[0]):
                for score in scores:
                    if "True" in key:
                        evaluation = evaluation_ecoli_opt_variants_dataframe[evaluation_ecoli_opt_variants_dataframe["variant"] == variant][f"{score_type}_evaluation_score_for_optimized_ecoli"]
                    else:
                        evaluation = evaluation_bacillus_opt_variants_dataframe[evaluation_bacillus_opt_variants_dataframe["variant"] == variant][f"{score_type}_evaluation_score_for_optimized_bacillus"]
                    evaluation = evaluation.reset_index(drop=True)
                    score_diff = evaluation - score
                    # print(score_diff)
                    if score_diff[0] <= 0 :
                        print(f"Found better random score: {score} for key: {key} and variant: {variant}")


fig, axes = plt.subplots(nrows=1,ncols=1, figsize=(15,10))

fig.suptitle("Variants Evaluation Z-Scores")
new_evaluation_ecoli_opt_variants_dataframe.plot("variant", "average_distance_zscore", ax=axes)
new_evaluation_bacillus_opt_variants_dataframe.plot("variant", "average_distance_zscore", ax=axes)
new_evaluation_ecoli_opt_variants_dataframe.plot("variant", "ratio_zscore", ax=axes, linestyle="dotted")
new_evaluation_bacillus_opt_variants_dataframe.plot("variant", "ratio_zscore", ax=axes, linestyle="dotted")
axes.legend(labels=["e_coli_optimized_average_distance_zscore", "bacillus_optimized_average_distance_zscore", "e_coli_optimized_ratio_zscore", "bacillus_optimized_ratio_zscore"])
axes.set_ylim([0, 20])


fig, axes = plt.subplots(nrows=1,ncols=2, figsize=(30,10))

fig.suptitle("Variants Evaluation Z-Scores")
new_evaluation_ecoli_opt_variants_dataframe.plot("variant", "average_distance_zscore", kind="barh", ax=axes[0], title="E.coli optimized")
new_evaluation_bacillus_opt_variants_dataframe.plot("variant", "average_distance_zscore", kind="barh", ax=axes[1], title="Bacillus optimized", color="#dc4d01")


fig, axes = plt.subplots(nrows=1,ncols=2, figsize=(30,10))

fig.suptitle("Variants Evaluation Z-Scores")
new_evaluation_ecoli_opt_variants_dataframe.plot("variant", "ratio_zscore", kind="barh", ax=axes[0], title="E.coli optimized")
new_evaluation_bacillus_opt_variants_dataframe.plot("variant", "ratio_zscore", kind="barh", ax=axes[1], title="Bacillus optimized", color="#dc4d01")


merged_df_ecoli = pd.merge(evaluation_ecoli_opt_variants_dataframe, ecoli_opt_variants_dataframe, on='variant', how='inner')
merged_df_ecoli['ecoli_freq_dist_diff'] =  merged_df_ecoli['ecoli_dist_for_optimized_ecoli'] - merged_df_ecoli['ecoli_dist_for_original_mcherry']
merged_df_ecoli['bacillus_freq_dist_diff'] = merged_df_ecoli['bacillus_dist_for_optimized_ecoli'] - merged_df_ecoli['bacillus_dist_for_original_mcherry']

merged_df_bacillus = pd.merge(evaluation_bacillus_opt_variants_dataframe, bacillus_opt_variants_dataframe, on='variant', how='inner')
merged_df_bacillus['ecoli_freq_dist_diff'] =  merged_df_bacillus['ecoli_dist_for_optimized_bacillus'] - merged_df_bacillus['ecoli_dist_for_original_mcherry']
merged_df_bacillus['bacillus_freq_dist_diff'] = merged_df_bacillus['bacillus_dist_for_optimized_bacillus'] - merged_df_bacillus['bacillus_dist_for_original_mcherry'] 


fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))
fig.suptitle("Ecaluation Score vs. Codon Frequencies distance")

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="ecoli_freq_dist_diff", ax=axes[0,0], title="E.coli optimized - Evaluation score vs.\nE.coli codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.spearmanr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["ecoli_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="bacillus_freq_dist_diff", ax=axes[0,1], title="E.coli optimized - Evaluation score vs.\nBacillus codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.spearmanr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["bacillus_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="ecoli_freq_dist_diff", ax=axes[1,0], title="Bacillus optimized - Evaluation score vs.\nE.coli codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.spearmanr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["ecoli_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="bacillus_freq_dist_diff", ax=axes[1,1], title="Bacillus optimized - Evaluation score vs.\nBacillus codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.spearmanr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["bacillus_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')


merged_df_ecoli = pd.merge(evaluation_ecoli_opt_greater_alpha_variants_dataframe, greater_alpha_ecoli_opt_variants_dataframe, on='variant', how='inner')
merged_df_ecoli['ecoli_freq_dist_diff'] =  merged_df_ecoli['ecoli_dist_for_optimized_ecoli'] - merged_df_ecoli['ecoli_dist_for_original_mcherry']
merged_df_ecoli['bacillus_freq_dist_diff'] = merged_df_ecoli['bacillus_dist_for_optimized_ecoli'] - merged_df_ecoli['bacillus_dist_for_original_mcherry']

merged_df_bacillus = pd.merge(evaluation_bacillus_opt_greater_alpha_variants_dataframe, greater_alpha_bacillus_opt_variants_dataframe, on='variant', how='inner')
merged_df_bacillus['ecoli_freq_dist_diff'] =  merged_df_bacillus['ecoli_dist_for_optimized_bacillus'] - merged_df_bacillus['ecoli_dist_for_original_mcherry']
merged_df_bacillus['bacillus_freq_dist_diff'] = merged_df_bacillus['bacillus_dist_for_optimized_bacillus'] - merged_df_bacillus['bacillus_dist_for_original_mcherry'] 


fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,13))
fig.suptitle("Ecaluation Score vs. Codon Frequencies distance (alpha=1)")

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="ecoli_freq_dist_diff", ax=axes[0,0], title="E.coli optimized - Evaluation score vs.\nE.coli codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["ecoli_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="bacillus_freq_dist_diff", ax=axes[0,1], title="E.coli optimized - Evaluation score vs.\nBacillus codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["bacillus_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="ecoli_freq_dist_diff", ax=axes[1,0], title="Bacillus optimized - Evaluation score vs.\nE.coli codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["ecoli_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="bacillus_freq_dist_diff", ax=axes[1,1], title="Bacillus optimized - Evaluation score vs.\nBacillus codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["bacillus_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')


merged_df_ecoli = pd.merge(evaluation_ecoli_opt_lower_alpha_variants_dataframe, lower_alpha_ecoli_opt_variants_dataframe, on='variant', how='inner')
merged_df_ecoli['ecoli_freq_dist_diff'] =  merged_df_ecoli['ecoli_dist_for_optimized_ecoli'] - merged_df_ecoli['ecoli_dist_for_original_mcherry']
merged_df_ecoli['bacillus_freq_dist_diff'] = merged_df_ecoli['bacillus_dist_for_optimized_ecoli'] - merged_df_ecoli['bacillus_dist_for_original_mcherry']

merged_df_bacillus = pd.merge(evaluation_bacillus_opt_lower_alpha_variants_dataframe, lower_alpha_bacillus_opt_variants_dataframe, on='variant', how='inner')
merged_df_bacillus['ecoli_freq_dist_diff'] =  merged_df_bacillus['ecoli_dist_for_optimized_bacillus'] - merged_df_bacillus['ecoli_dist_for_original_mcherry']
merged_df_bacillus['bacillus_freq_dist_diff'] = merged_df_bacillus['bacillus_dist_for_optimized_bacillus'] - merged_df_bacillus['bacillus_dist_for_original_mcherry'] 


fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,13))
fig.suptitle("Ecaluation Score vs. Codon Frequencies distance (alpha=0)")

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="ecoli_freq_dist_diff", ax=axes[0,0], title="Bacillus deoptimized - Evaluation score vs.\nE.coli codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["ecoli_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="bacillus_freq_dist_diff", ax=axes[0,1], title="Bacillus deoptimized - Evaluation score vs.\nBacillus codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["bacillus_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="ecoli_freq_dist_diff", ax=axes[1,0], title="E.coli deoptimized - Evaluation score vs.\nE.coli codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["ecoli_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="bacillus_freq_dist_diff", ax=axes[1,1], title="E.coli deoptimized - Evaluation score vs.\nBacillus codon frequencies diff (modified seq dist-original seq dist)")
r, p = stats.pearsonr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["bacillus_freq_dist_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')


merged_df_ecoli = pd.merge(evaluation_ecoli_opt_variants_dataframe, cub_ecoli_opt_variants_dataframe, on='variant', how='inner')
merged_df_ecoli['ecoli_score_diff'] = merged_df_ecoli['ecoli_final_score'] - merged_df_ecoli['ecoli_initial_score']
merged_df_ecoli['bacillus_score_diff'] = merged_df_ecoli['bacillus_final_score'] - merged_df_ecoli['bacillus_initial_score']

merged_df_bacillus = pd.merge(evaluation_bacillus_opt_variants_dataframe, cub_bacillus_opt_variants_dataframe, on='variant', how='inner')
merged_df_bacillus['ecoli_score_diff'] = merged_df_bacillus['ecoli_final_score'] - merged_df_bacillus['ecoli_initial_score']
merged_df_bacillus['bacillus_score_diff'] = merged_df_bacillus['bacillus_final_score'] - merged_df_bacillus['bacillus_initial_score']

fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="ecoli_score_diff", ax=axes[0,0], title="E.coli optimized - Evaluation score vs.\nE.coli CUB score diff (modified seq score-original seq score)")
r, p = stats.spearmanr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["ecoli_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_ecoli.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="bacillus_score_diff", ax=axes[0,1], title="E.coli optimized - Evaluation score vs.\nBacillus CUB score diff (modified seq score-original seq score)")
r, p = stats.spearmanr(merged_df_ecoli["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli["bacillus_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="ecoli_score_diff", ax=axes[1,0], title="Bacillus optimized - Evaluation score vs.\nE.coli CUB score diff (modified seq score-original seq score)")
r, p = stats.spearmanr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["ecoli_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="bacillus_score_diff", ax=axes[1,1], title="Bacillus optimized - Evaluation score vs.\nBacillus CUB score diff (modified seq score-original seq score)")
r, p = stats.spearmanr(merged_df_bacillus["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus["bacillus_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')


merged_df_ecoli_greater_alpha = pd.merge(evaluation_ecoli_opt_greater_alpha_variants_dataframe, cub_ecoli_opt_greater_alpha_variants_dataframe, on='variant', how='inner')
merged_df_ecoli_greater_alpha['ecoli_score_diff'] = merged_df_ecoli_greater_alpha['ecoli_final_score'] - merged_df_ecoli_greater_alpha['ecoli_initial_score']
merged_df_ecoli_greater_alpha['bacillus_score_diff'] = merged_df_ecoli_greater_alpha['bacillus_final_score'] - merged_df_ecoli_greater_alpha['bacillus_initial_score']

merged_df_bacillus_greater_alpha = pd.merge(evaluation_bacillus_opt_greater_alpha_variants_dataframe, cub_bacillus_opt_greater_alpha_variants_dataframe, on='variant', how='inner')
merged_df_bacillus_greater_alpha['ecoli_score_diff'] = merged_df_bacillus_greater_alpha['ecoli_final_score'] - merged_df_bacillus_greater_alpha['ecoli_initial_score']
merged_df_bacillus_greater_alpha['bacillus_score_diff'] = merged_df_bacillus_greater_alpha['bacillus_final_score'] - merged_df_bacillus_greater_alpha['bacillus_initial_score']

fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))

scatter = merged_df_ecoli_greater_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="ecoli_score_diff", ax=axes[0,0], title="E.coli optimized (alpha=1)- Evaluation score vs.\nE.coli CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_ecoli_greater_alpha["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli_greater_alpha["ecoli_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_ecoli_greater_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="bacillus_score_diff", ax=axes[0,1], title="E.coli optimized (alpha=1)- Evaluation score vs.\nBacillus CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_ecoli_greater_alpha["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli_greater_alpha["bacillus_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus_greater_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="ecoli_score_diff", ax=axes[1,0], title="Bacillus optimized (alpha=1)- Evaluation score vs.\nE.coli CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_bacillus_greater_alpha["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus_greater_alpha["ecoli_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus_greater_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="bacillus_score_diff", ax=axes[1,1], title="Bacillus optimized (alpha=1)- Evaluation score vs.\nBacillus CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_bacillus_greater_alpha["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus_greater_alpha["bacillus_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')


merged_df_ecoli_lower_alpha = pd.merge(evaluation_ecoli_opt_lower_alpha_variants_dataframe, cub_ecoli_opt_lower_alpha_variants_dataframe, on='variant', how='inner')
merged_df_ecoli_lower_alpha['ecoli_score_diff'] = merged_df_ecoli_lower_alpha['ecoli_final_score'] - merged_df_ecoli_lower_alpha['ecoli_initial_score']
merged_df_ecoli_lower_alpha['bacillus_score_diff'] = merged_df_ecoli_lower_alpha['bacillus_final_score'] - merged_df_ecoli_lower_alpha['bacillus_initial_score']

merged_df_bacillus_lower_alpha = pd.merge(evaluation_bacillus_opt_lower_alpha_variants_dataframe, cub_bacillus_opt_lower_alpha_variants_dataframe, on='variant', how='inner')
merged_df_bacillus_lower_alpha['ecoli_score_diff'] = merged_df_bacillus_lower_alpha['ecoli_final_score'] - merged_df_bacillus_lower_alpha['ecoli_initial_score']
merged_df_bacillus_lower_alpha['bacillus_score_diff'] = merged_df_bacillus_lower_alpha['bacillus_final_score'] - merged_df_bacillus_lower_alpha['bacillus_initial_score']

fig, axes = plt.subplots(nrows=2,ncols=2, figsize=(20,15))

scatter = merged_df_ecoli_lower_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="ecoli_score_diff", ax=axes[0,0], title="Bacillus deoptimized (alpha=0)- Evaluation score vs.\nE.coli CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_ecoli_lower_alpha["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli_lower_alpha["ecoli_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_ecoli_lower_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_ecoli", y="bacillus_score_diff", ax=axes[0,1], title="Bacillus deoptimized (alpha=0)- Evaluation score vs.\nBacillus CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_ecoli_lower_alpha["average_distance_evaluation_score_for_optimized_ecoli"], merged_df_ecoli_lower_alpha["bacillus_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus_lower_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="ecoli_score_diff", ax=axes[1,0], title="E.coli deoptimized (alpha=0)- Evaluation score vs.\nE.coli CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_bacillus_lower_alpha["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus_lower_alpha["ecoli_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')

scatter = merged_df_bacillus_lower_alpha.plot.scatter(x="average_distance_evaluation_score_for_optimized_bacillus", y="bacillus_score_diff", ax=axes[1,1], title="E.coli deoptimized (alpha=0)- Evaluation score vs.\nBacillus CUB score diff (modified seq score-original seq score")
r, p = stats.spearmanr(merged_df_bacillus_lower_alpha["average_distance_evaluation_score_for_optimized_bacillus"], merged_df_bacillus_lower_alpha["bacillus_score_diff"])
scatter.annotate('r = {:.3f}'.format(r), xy=(0.7, 0.9), xycoords='axes fraction')


my_colors = ['#3b5b92', '#3b5b92', '#dc4d01', '#dc4d01'] 
my_colors_extended = ['cornflowerblue', 'cornflowerblue', 'navy', 'navy', 'orange', 'orange', 'orangered', 'orangered'] 
for q in range(cub_ecoli_opt_greater_alpha_variants_dataframe.shape[0]):

    fig, axes = plt.subplots(nrows=1,ncols=3, figsize=(20,5))
    fig.suptitle(greater_alpha_ecoli_opt_variants_dataframe.iloc[q,0] + " - codon frequency distance")
    lower_alpha_ecoli_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[0], stacked=True, color=my_colors_extended)
    axes[0].set_title("alpha = 0")
    ecoli_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[1], stacked=True, color=my_colors_extended)
    axes[1].set_title("alpha = 0.5")
    greater_alpha_ecoli_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[2], stacked=True, color=my_colors_extended)
    axes[2].set_title("alpha = 1")
    
    fig, axes = plt.subplots(nrows=1,ncols=3, figsize=(20,5))
    fig.suptitle(greater_alpha_ecoli_opt_variants_dataframe_highly_expressed.iloc[q,0] + " - codon frequency distance (highly expressed)")
    lower_alpha_ecoli_opt_variants_dataframe_highly_expressed.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[0], stacked=True, color=my_colors_extended)
    axes[0].set_title("alpha = 0")
    ecoli_opt_variants_dataframe_highly_expressed.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[1], stacked=True, color=my_colors_extended)
    axes[1].set_title("alpha = 0.5")
    greater_alpha_ecoli_opt_variants_dataframe_highly_expressed.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[2], stacked=True, color=my_colors_extended)
    axes[2].set_title("alpha = 1")


    fig, axes = plt.subplots(nrows=1,ncols=3, figsize=(20,5))
    fig.suptitle(cub_ecoli_opt_greater_alpha_variants_dataframe.iloc[q,0] + " - CUB scores")
    cub_ecoli_opt_lower_alpha_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[0], stacked=True, color=my_colors)
    axes[0].set_title("alpha = 0")
    cub_ecoli_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[1], stacked=True, color=my_colors)
    axes[1].set_title("alpha = 0.5")
    cub_ecoli_opt_greater_alpha_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[2], stacked=True, color=my_colors)
    axes[2].set_title("alpha = 1")





my_colors = ['#3b5b92', '#3b5b92', '#dc4d01', '#dc4d01'] 
my_colors_extended = ['cornflowerblue', 'cornflowerblue', 'navy', 'navy', 'orange', 'orange', 'orangered', 'orangered'] 
for q in range(cub_bacillus_opt_greater_alpha_variants_dataframe.shape[0]):

    fig, axes = plt.subplots(nrows=1,ncols=3, figsize=(20,5))
    fig.suptitle(greater_alpha_bacillus_opt_variants_dataframe.iloc[q,0] + " - codon frequency distance")
    lower_alpha_bacillus_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[0], stacked=True, color=my_colors_extended)
    axes[0].set_title("alpha = 0")
    bacillus_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[1], stacked=True, color=my_colors_extended)
    axes[1].set_title("alpha = 0.5")
    greater_alpha_bacillus_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[2], stacked=True, color=my_colors_extended)
    axes[2].set_title("alpha = 1")
    
    fig, axes = plt.subplots(nrows=1,ncols=3, figsize=(20,5))
    fig.suptitle(greater_alpha_bacillus_opt_variants_dataframe_highly_expressed.iloc[q,0] + " - codon frequency distance (highly expressed)")
    lower_alpha_bacillus_opt_variants_dataframe_highly_expressed.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[0], stacked=True, color=my_colors_extended)
    axes[0].set_title("alpha = 0")
    bacillus_opt_variants_dataframe_highly_expressed.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[1], stacked=True, color=my_colors_extended)
    axes[1].set_title("alpha = 0.5")
    greater_alpha_bacillus_opt_variants_dataframe_highly_expressed.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[2], stacked=True, color=my_colors_extended)
    axes[2].set_title("alpha = 1")


    fig, axes = plt.subplots(nrows=1,ncols=3, figsize=(20,5))
    fig.suptitle(cub_bacillus_opt_greater_alpha_variants_dataframe.iloc[q,0] + " - CUB scores")
    cub_bacillus_opt_lower_alpha_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[0], stacked=True, color=my_colors)
    axes[0].set_title("alpha = 0")
    cub_bacillus_opt_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[1], stacked=True, color=my_colors)
    axes[1].set_title("alpha = 0.5")
    cub_bacillus_opt_greater_alpha_variants_dataframe.iloc[q,1:].plot(x="variant", kind="bar", ax=axes[2], stacked=True, color=my_colors)
    axes[2].set_title("alpha = 1")


mcherry_ecoli_dist_scores = [-5.350772658,-1.302566563,-5.308514706,
0.218293345,
-4.085662496,
-1.879319816,
-4.200511895,
-3.358996535,
]

mcherry_bacillus_dist_scores = [-1.432549184,
-5.109467844,
-0.84180732,
-3.596564271,
3.878153094,
-6.619242448,
3.754522929,
-7.763015693,
]

average_distance_scores = [
    1.959111737,
1.90345064,
2.233353693,
1.907428808,
3.981907795,
2.369961316,
3.977517412,
2.202009579,
]

optimization_methods = ["single_codon_diff_bacillus_optimized",
"single_codon_diff_ecoli_optimized",
"single_codon_ratio_bacillus_optimized",
"single_codon_ratio_ecoli_optimized",
"zscore_bulk_aa_diff_bacillus_optimized",
"zscore_bulk_aa_diff_ecoli_optimized",
"zscore_bulk_aa_ratio_bacillus_optimized",
"zscore_bulk_aa_ratio_ecoli_optimized",
]

mcherry_dataframe_dict = {
    "mcherry_ecoli_dist_scores": mcherry_ecoli_dist_scores,
    "mcherry_bacillus_dist_scores": mcherry_bacillus_dist_scores,
    "average_distance_scores": average_distance_scores,
    "variant": optimization_methods,
}


mcherry_df = pd.DataFrame(mcherry_dataframe_dict)



fig, axes = plt.subplots(figsize=(20,5))
mcherry_df.plot(x="variant", kind="bar", ax=axes, title="mcherry evaluation scores vs. dist scores")





root_dir = r"C:\projects\Igem_TAU_2021_moran\analysis\orf_model_analysis\results\endogenous-remote-debug\CAI_single_codon_diff_ecoli_opt_False_DppD"
run_summary_path = os.path.join(root_dir, "run_summary.json")

with open(run_summary_path, "r") as json_file:
    run_summary = json.load(json_file)

stats_dict = defaultdict(list)
for codon in run_summary["orf_sequence_stats"].keys():
    stats_dict["codon"].append(codon)
    stats_dict["aa"].append(shared_functions_and_vars.nt_to_aa[codon])
    stats_dict["counts"].append(run_summary["orf_sequence_stats"][codon])
    stats_dict["optimized_loss"].append(run_summary["orf_debug"]["optimized_loss"][shared_functions_and_vars.nt_to_aa[codon]][codon])
    stats_dict["deoptimized_loss"].append(run_summary["orf_debug"]["deoptimized_loss"][shared_functions_and_vars.nt_to_aa[codon]][codon])
    stats_dict["total_loss"].append(run_summary["orf_debug"]["total_loss"][shared_functions_and_vars.nt_to_aa[codon]][codon])
    stats_dict["optimal"].append(run_summary["orf"]["aa_to_optimal_codon"][shared_functions_and_vars.nt_to_aa[codon]] == codon)

stats_df = pd.DataFrame(stats_dict)
total = stats_df.groupby(stats_df["aa"]).aggregate({"counts": "sum"})

filtered_df = stats_df[stats_df.optimal]
merged = filtered_df.merge(total, on='aa', how='inner', suffixes=("", "_aa") )
merged


# scatter = merged_df_ecoli_lower_alpha.plot.scatter(x="evaluation_score_for_optimized_ecoli", y="ecoli_score_diff", ax=axes[0,0], title="Bacillus deoptimized (alpha=0)- Evaluation score vs.\nE.coli CUB score diff (modified seq score-original seq score")
# TODO - think how to visualize the data in order to show the trend
# stats_df.plot()

colors = ['r', 'b', 'k', 'g', 'm']
merged.plot.scatter(x="optimized_loss", y="deoptimized_loss", s=10*merged.counts_aa)
stats_df.plot.scatter(x="optimized_loss", y="deoptimized_loss", s=10*stats_df.counts)


import codonbias as cb
nuc = cb.stats.BaseCounter()
variants_to_check = {}
variants_to_check["original"] = original_mcherry_seq
variants_to_check["5_zscore_diff_ecoli_opt_false"] = "ATGGTTTCAAAAGGGGAGGAGGACAATATGGCAATTATCAAAGAATTTATGAGATTTAAAGTTCATATGGAAGGATCAGTTAACGGACATGAATTTGAAATTGAAGGAGAAGGAGAAGGAAGACCTTATGAAGGAACACAAACAGCAAAACTTAAAGTTACAAAAGGAGGCCCTCTTCCTTTTGCATGGGATATTCTTTCACCTCAATTTATGTATGGATCAAAAGCATATGTTAAACATCCTGCAGATATTCCTGATTATCTTAAACTTTCATTTCCTGAAGGATTTAAATGGGAAAGAGTTATGAACTTTGAAGATGGAGGCGTTGTAACAGTTACACAAGATTCATCTCTTCAAGATGGAGAATTTATTTATAAAGTTAAACTTAGAGGAACAAACTTTCCTTCAGATGGACCTGTTATGCAAAAAAAGACAATGGGATGGGAAGCATCATCTGAAAGAATGTATCCTGAAGATGGAGCACTTAAAGGAGAAATTAAACAAAGACTTAAACTTAAAGATGGAGGCCATTATGATGCAGAAGTTAAAACAACGTATAAAGCAAAAAAGCCTGTTCAACTTCCTGGAGCATATAACGTTAACATTAAACTTGATATTACATCACATAACGAAGATTATACAATTGTTGAACAATATGAAAGAGCAGAAGGAAGACATTCAACAGGAGGCATGGATGAACTTTATAAATAA"
variants_to_check["7_zscore_ratio_ecoli_opt_false"] = "ATGGTTTCAAAAGGGGAGGAGGACAATATGGCAATTATCAAAGAATTTATGAGATTTAAAGTTCATATGGAAGGATCAGTTAATGGACATGAATTTGAAATTGAAGGAGAAGGAGAAGGAAGACCTTATGAAGGAACACAAACAGCAAAACTTAAAGTTACAAAAGGAGGCCCTCTTCCTTTTGCATGGGATATTCTTTCACCTCAATTTATGTATGGATCAAAAGCATATGTTAAACATCCTGCAGATATTCCTGATTATCTTAAACTTTCATTTCCTGAAGGATTTAAATGGGAAAGAGTTATGAATTTTGAAGATGGAGGCGTTGTAACAGTTACACAAGATTCATCTCTTCAAGATGGAGAATTTATTTATAAAGTTAAACTTAGAGGAACAAATTTTCCTTCAGATGGACCTGTTATGCAAAAAAAGACAATGGGATGGGAAGCATCATCTGAAAGAATGTATCCTGAAGATGGAGCACTTAAAGGAGAAATTAAACAAAGACTTAAACTTAAAGATGGAGGCCATTATGATGCAGAAGTTAAAACAACGTATAAAGCAAAAAAGCCTGTTCAACTTCCTGGAGCATATAATGTTAATATTAAACTTGATATTACATCACATAATGAAGATTATACAATTGTTGAACAATATGAAAGAGCAGAAGGAAGACATTCAACAGGAGGCATGGATGAACTTTATAAATAA"
variants_to_check["3_single_ratio_ecoli_opt_false"] = "ATGGTTTCAAAAGGGGAGGAGGACAATATGGCAATTATCAAAGAATTTATGAGATTTAAGGTACATATGGAGGGATCAGTAAATGGACATGAGTTTGAGATTGAGGGAGAGGGAGAGGGAAGACCTTATGAGGGAACACAAACAGCAAAGCTTAAGGTAACAAAGGGAGGGCCTCTTCCTTTTGCATGGGATATTCTTTCACCTCAATTTATGTATGGATCAAAGGCATATGTAAAGCATCCTGCAGATATTCCTGATTATCTTAAGCTTTCATTTCCTGAGGGATTTAAGTGGGAGAGAGTAATGAATTTTGAGGATGGAGGGGTAGTCACAGTAACACAAGATTCATCTCTTCAAGATGGAGAGTTTATTTATAAGGTAAAGCTTAGAGGAACAAATTTTCCTTCAGATGGACCTGTAATGCAAAAGAAAACAATGGGATGGGAGGCATCATCTGAGAGAATGTATCCTGAGGATGGAGCACTTAAGGGAGAGATTAAGCAAAGACTTAAGCTTAAGGATGGAGGGCATTATGATGCAGAGGTAAAGACAACGTATAAGGCAAAGAAACCTGTACAACTTCCTGGAGCATATAATGTAAATATTAAGCTTGATATTACATCACATAATGAGGATTATACAATTGTAGAGCAATATGAGAGAGCAGAGGGAAGACATTCAACAGGAGGGATGGATGAGCTTTATAAGTAG"

for variant_name, variant_seq in variants_to_check.items():
    print("checking " + variant_name)
    freq = nuc.count(variant_seq).get_table(normed=True)
    gc_content = freq['G'] + freq['C']
    print(f"GC content: {gc_content}")
    prev_codon = None
    repeating_seq_len = 0
    repeating_index = None
    for i in range(0, len(variant_seq), 3):
        codon = variant_seq[i:i+3]
        if codon == prev_codon:
            repeating_seq_len += 1
            if repeating_index is None:
                repeating_index = i
        else:
            if repeating_seq_len != 0:
                print(f"found repeating of codon {prev_codon} of length {repeating_seq_len} starting at index {repeating_index}")
            repeating_seq_len = 0
            repeating_index = None
        prev_codon = codon
        





import codonbias as cb
nuc = cb.stats.BaseCounter()


variants_to_check = {}
variants_to_check["5_zscore_diff_ecoli_opt_false"] = "ATGGTTTCAAAAGGGGAGGAGGACAATATGGCAATTATCAAAGAATTTATGAGATTTAAAGTTCATATGGAAGGATCAGTTAACGGACATGAATTTGAAATTGAAGGAGAAGGAGAAGGAAGACCTTATGAAGGAACACAAACAGCAAAACTTAAAGTTACAAAAGGAGGCCCTCTTCCTTTTGCATGGGATATTCTTTCACCTCAATTTATGTATGGATCAAAAGCATATGTTAAACATCCTGCAGATATTCCTGATTATCTTAAACTTTCATTTCCTGAAGGATTTAAATGGGAAAGAGTTATGAACTTTGAAGATGGAGGCGTTGTAACAGTTACACAAGATTCATCTCTTCAAGATGGAGAATTTATTTATAAAGTTAAACTTAGAGGAACAAACTTTCCTTCAGATGGACCTGTTATGCAAAAAAAGACAATGGGATGGGAAGCATCATCTGAAAGAATGTATCCTGAAGATGGAGCACTTAAAGGAGAAATTAAACAAAGACTTAAACTTAAAGATGGAGGCCATTATGATGCAGAAGTTAAAACAACGTATAAAGCAAAAAAGCCTGTTCAACTTCCTGGAGCATATAACGTTAACATTAAACTTGATATTACATCACATAACGAAGATTATACAATTGTTGAACAATATGAAAGAGCAGAAGGAAGACATTCAACAGGAGGCATGGATGAACTTTATAAATAA"
variants_to_check["7_zscore_ratio_ecoli_opt_false"] = "ATGGTTTCAAAAGGGGAGGAGGACAATATGGCAATTATCAAAGAATTTATGAGATTTAAAGTTCATATGGAAGGATCAGTTAATGGACATGAATTTGAAATTGAAGGAGAAGGAGAAGGAAGACCTTATGAAGGAACACAAACAGCAAAACTTAAAGTTACAAAAGGAGGCCCTCTTCCTTTTGCATGGGATATTCTTTCACCTCAATTTATGTATGGATCAAAAGCATATGTTAAACATCCTGCAGATATTCCTGATTATCTTAAACTTTCATTTCCTGAAGGATTTAAATGGGAAAGAGTTATGAATTTTGAAGATGGAGGCGTTGTAACAGTTACACAAGATTCATCTCTTCAAGATGGAGAATTTATTTATAAAGTTAAACTTAGAGGAACAAATTTTCCTTCAGATGGACCTGTTATGCAAAAAAAGACAATGGGATGGGAAGCATCATCTGAAAGAATGTATCCTGAAGATGGAGCACTTAAAGGAGAAATTAAACAAAGACTTAAACTTAAAGATGGAGGCCATTATGATGCAGAAGTTAAAACAACGTATAAAGCAAAAAAGCCTGTTCAACTTCCTGGAGCATATAATGTTAATATTAAACTTGATATTACATCACATAATGAAGATTATACAATTGTTGAACAATATGAAAGAGCAGAAGGAAGACATTCAACAGGAGGCATGGATGAACTTTATAAATAA"



def find_closest_codon_to_index(index_list, variant):
    for bad_index in index_list:
        idt_start_index = bad_index
        start_index = idt_start_index - 1
        # if variant[start_index:start_index+5] != "ATATG":
            # print (f"bad seq: {variant[start_index:start_index+5]} at index {start_index}")
        nearest_start_codon_idx = find_nearest(codons_indices, start_index)
        nearest_start_codon = codons_indices[nearest_start_codon_idx]
        print(f"nearest codon to bad index {start_index} start at {nearest_start_codon} and is {variant[nearest_start_codon:nearest_start_codon+3]}") 


def calculate_gc_content(seq):
    freq = nuc.count(seq).get_table(normed=True)
    gc_content = freq['G'] + freq['C']
    print(f"GC content: {gc_content}")





variant_5 = variants_to_check["5_zscore_diff_ecoli_opt_false"] 
codons_indices = list(range(0,len(variant),3))


aa_to_selected_codon: {
            "C": "TGC",
            "D": "GAT",
            "S": "TCA",
            "Q": "CAA",
            "M": "ATG",
            "N": "AAC",
            "P": "CCT",
            "K": "AAA",
            "_": "TAA",
            "T": "ACA",
            "F": "TTT",
            "A": "GCA",
            "G": "GGA",
            "I": "ATT",
            "L": "CTT",
            "H": "CAT",
            "R": "AGA",
            "W": "TGG",
            "V": "GTT",
            "E": "GAA",
            "Y": "TAT"
}


aa_to_default_codon = {
            "C": "TGT",
            "D": "GAC",
            "S": "TCT",
            "Q": "CAG",
            "M": None,
            "N": "AAT",
            "P": "CCG",
            "K": "AAG",
            "_": "TGA",
            "T": "ACG",
            "F": "TTC",
            "A": "GCT",
            "G": "GGC",
            "I": "ATC",
            "L": "TTA",
            "H": "CAC",
            "R": "CGC",
            "W": None,
            "V": "GTA",
            "E": "GAG",
            "Y": "TAC"
}


# The repeated sequence ATATG exists at the following locations: 26, 65, 228, 654, 64, 227, 590
repeated_seq_idxs = [26, 65, 228, 654, 64, 227, 590]
find_closest_codon_to_index(repeated_seq_idxs, variant_5)


# The repeated sequence AACTTTATAAATAA exists at the following locations: 698, 368.Solution: Redesign to disrupt instances of the repeat near the terminal end(s) or add extra bases to push the repeat internally.
repeated_seq_idxs = [368, 698]
find_closest_codon_to_index(repeated_seq_idxs, variant_5)


variant_5[225:225+3]


new_variant = variant_5
index = 63
assert index in codons_indices
new_variant = new_variant[:index]+ "CAC" + new_variant[index+3:]
assert len(new_variant) == 711
index = 225 
assert index in codons_indices
new_variant = new_variant[:index]+ "GCT" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 651
assert index in codons_indices
new_variant = new_variant[:index]+ "CAG" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 366
assert index in codons_indices
new_variant = new_variant[:index]+ "TTC" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 696
assert index in codons_indices
new_variant = new_variant[:index]+ "GAG" + new_variant[index+3:] 
assert len(new_variant) == 711

# These manual changes aim to mitigate the low GC content around key indices mentioned by the scoring system
index = 591
assert index in codons_indices
new_variant = new_variant[:index]+ "TAC" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 207
assert index in codons_indices
new_variant = new_variant[:index]+ "TTC" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 558
assert index in codons_indices
new_variant = new_variant[:index]+ "AAG" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 264
assert index in codons_indices
new_variant = new_variant[:index]+ "AAG" + new_variant[index+3:] 
assert len(new_variant) == 711
new_variant


optimization_method = "zscore_bulk_aa_diff"
optimization_cub_index = "CAI"
is_ecoli_optimized = False
dummy_output_path = "results/mcherry_debug"

user_input = get_user_input(
    optimization_method=optimization_method,
    optimization_cub_index=optimization_cub_index,
    is_ecoli_optimized=is_ecoli_optimized,
    orf_sequence=str(original_mcherry_seq),
    output_path=dummy_output_path,
    should_use_protein_abundance=True,
)
variant_5_evaluation_score = evaluate_score_for_sequence(
    sequence=variant_5,
    user_input=user_input,
    optimization_method=optimization_method,
    optimization_cub_index=optimization_cub_index,
    is_ecoli_optimized=is_ecoli_optimized,
    orf_sequence=original_mcherry_seq,
    output_path=dummy_output_path,
)
new_variant_5_evaluation_score = evaluate_score_for_sequence(
    sequence=new_variant,
    user_input=user_input,
    optimization_method=optimization_method,
    optimization_cub_index=optimization_cub_index,
    is_ecoli_optimized=is_ecoli_optimized,
    orf_sequence=original_mcherry_seq,
    output_path=dummy_output_path,
)
print(f"Evaluation score changed to {new_variant_5_evaluation_score.average_distance_score} compared to {variant_evaluation_score.average_distance_score}")


original_score = 47.9
new_score = 9.6
print(f"Original sequence gBlocks score is {original_score}. \nNew sequence gBlocks score is {new_score}")


find_closest_codon_to_index([545], variant_5)


offset = 189
for i in range(0, 100, 3):
    print(i+offset, new_variant[i+offset:i+offset+3])





new_variant = ""
changed_codon_count = 0
for i in range(0, len(variant), 3):
    codon = variant[i:i+3]
    new_codon = codon
    if codon == "AAA":
        new_codon = "AAG"
        changed_codon_count += 1
    new_variant += new_codon


user_input = get_user_input(
    optimization_method=optimization_method,
    optimization_cub_index=optimization_cub_index,
    is_ecoli_optimized=is_ecoli_optimized,
    orf_sequence=str(original_mcherry_seq),
    output_path=dummy_output_path,
    should_use_protein_abundance=True,
)
new_variant_5_evaluation_score = evaluate_score_for_sequence(
    sequence=new_variant,
    user_input=user_input,
    optimization_method=optimization_method,
    optimization_cub_index=optimization_cub_index,
    is_ecoli_optimized=is_ecoli_optimized,
    orf_sequence=original_mcherry_seq,
    output_path=dummy_output_path,
)
print(f"Number of replaced codons is {changed_codon_count} out of {711/3}")
print(f"Evaluation score changed to {new_variant_5_evaluation_score.average_distance_score} compared to {variant_evaluation_score.average_distance_score}")





variant_7 = variants_to_check["7_zscore_ratio_ecoli_opt_false"] 
codons_indices = list(range(0,len(variant),3))


aa_to_selected_codon = {
            "C": "TGC",
            "D": "GAT",
            "S": "TCA",
            "Q": "CAA",
            "M": "ATG",
            "N": "AAT",
            "P": "CCT",
            "K": "AAA",
            "_": "TAA",
            "T": "ACA",
            "F": "TTT",
            "A": "GCA",
            "G": "GGA",
            "I": "ATT",
            "L": "CTT",
            "H": "CAT",
            "R": "AGA",
            "W": "TGG",
            "V": "GTT",
            "E": "GAA",
            "Y": "TAT"
}


aa_to_default_codon = {
            "C": "TGT",
            "D": "GAC",
            "S": "TCT",
            "Q": "CAG",
            "M": None,
            "N": "AAC",
            "P": "CCG",
            "K": "AAG",
            "_": "TAG",
            "T": "ACG",
            "F": "TTC",
            "A": "GCT",
            "G": "GGC",
            "I": "ATC",
            "L": "TTA",
            "H": "CAC",
            "R": "CGC",
            "W": None,
            "V": "GTA",
            "E": "GAG",
            "Y": "TAC",
}


# The repeated sequence ATATG exists at the following locations: 26, 65, 228, 654, 64, 227, 590.
repeated_seq_idxs = [26, 65, 228, 654, 64, 227, 590]
find_closest_codon_to_index(repeated_seq_idxs, variant_7)


new_variant = variant_7
index = 63
assert index in codons_indices
new_variant = new_variant[:index]+ "CAC" + new_variant[index+3:]
assert len(new_variant) == 711
index = 225
assert index in codons_indices
new_variant = new_variant[:index]+ "GCT" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 651
assert index in codons_indices
new_variant = new_variant[:index]+ "CAG" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 366
assert index in codons_indices
new_variant = new_variant[:index]+ "TTC" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 696
assert index in codons_indices
new_variant = new_variant[:index]+ "GAG" + new_variant[index+3:] 
assert len(new_variant) == 711

# These manual changes aim to mitigate the low GC content around key indices mentioned by the scoring system
index = 288  # 0.4
assert index in codons_indices
new_variant = new_variant[:index]+ "GGC" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 192 # 0.3
assert index in codons_indices
new_variant = new_variant[:index]+ "ATC" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 585
assert index in codons_indices
new_variant = new_variant[:index]+ "CCG" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 564 # 1.3
assert index in codons_indices
new_variant = new_variant[:index]+ "AAG" + new_variant[index+3:] 
assert len(new_variant) == 711
index = 600 # 0.6
assert index in codons_indices
new_variant = new_variant[:index]+ "AAC" + new_variant[index+3:] 
assert len(new_variant) == 711 
index = 576
assert index in codons_indices
new_variant = new_variant[:index]+ "CAG" + new_variant[index+3:] 
assert len(new_variant) == 711
new_variant


user_input = get_user_input(
    optimization_method=optimization_method,
    optimization_cub_index=optimization_cub_index,
    is_ecoli_optimized=is_ecoli_optimized,
    orf_sequence=str(original_mcherry_seq),
    output_path=dummy_output_path,
    should_use_protein_abundance=True,
)
new_variant_7_evaluation_score = evaluate_score_for_sequence(
    sequence=new_variant,
    user_input=user_input,
    optimization_method=optimization_method,
    optimization_cub_index=optimization_cub_index,
    is_ecoli_optimized=is_ecoli_optimized,
    orf_sequence=original_mcherry_seq,
    output_path=dummy_output_path,
)
print(f"Evaluation score changed to {new_variant_7_evaluation_score.average_distance_score} compared to {variant_evaluation_score.average_distance_score}")


original_score = 50.4
new_score = 9.3
print(f"Original sequence gBlocks score is {original_score}. New gBlocks score is {new_score}")


repeated_seq_idxs = [368, 698]
find_closest_codon_to_index(repeated_seq_idxs, variant_7)


gc_sensitive_indices = [545]
find_closest_codon_to_index(gc_sensitive_indices, variant_7)


offset = 543
for i in range(0, 100, 3):
    print(i+offset, new_variant[i+offset:i+offset+3])





import csv
homolog_prefix = "CATGCACGAATAAATTAGCAATAAAAGGAGGTTACTCGAGC"
homolog_suffix = "ggtactcgaggcatcaaataaaacgaaaggctcagtc"


file_name = "/Users/mkama/Documents/thesis/mcherry in-vivo experiments/mcherry_variants.csv"
with open(file_name, newline='') as csvfile:
    csv_reader = csv.reader(csvfile, delimiter=',', quotechar='|')
    i = -1
    for row in csv_reader:
        i += 1
        if i == 0 or len(row) < 2:
            continue
        variant = row[1]
        print(i/2, row[0])
        print("****")
        modified_variant = homolog_prefix + variant + homolog_suffix
        assert len(modified_variant) == 789
        print(modified_variant.upper())


manual_5 = "ATGGTTTCAAAAGGGGAGGAGGACAATATGGCAATTATCAAAGAATTTATGAGATTTAAAGTTCACATGGAAGGATCAGTTAACGGACATGAATTTGAAATTGAAGGAGAAGGAGAAGGAAGACCTTATGAAGGAACACAAACAGCAAAACTTAAAGTTACAAAAGGAGGCCCTCTTCCTTTTGCATGGGATATTCTTTCACCTCAATTCATGTATGGATCAAAAGCTTATGTTAAACATCCTGCAGATATTCCTGATTATCTTAAGCTTTCATTTCCTGAAGGATTTAAATGGGAAAGAGTTATGAACTTTGAAGATGGAGGCGTTGTAACAGTTACACAAGATTCATCTCTTCAAGATGGAGAATTCATTTATAAAGTTAAACTTAGAGGAACAAACTTTCCTTCAGATGGACCTGTTATGCAAAAAAAGACAATGGGATGGGAAGCATCATCTGAAAGAATGTATCCTGAAGATGGAGCACTTAAAGGAGAAATTAAACAAAGACTTAAACTTAAAGATGGAGGCCATTATGATGCAGAAGTTAAAACAACGTATAAGGCAAAAAAGCCTGTTCAACTTCCTGGAGCATACAACGTTAACATTAAACTTGATATTACATCACATAACGAAGATTATACAATTGTTGAACAGTATGAAAGAGCAGAAGGAAGACATTCAACAGGAGGCATGGATGAGCTTTATAAATAA"

complete_5 = homolog_prefix + manual_5 + homolog_suffix
assert len(complete_5) == 789
complete_5.upper()



manual_7 = "ATGGTTTCAAAAGGGGAGGAGGACAATATGGCAATTATCAAAGAATTTATGAGATTTAAAGTTCACATGGAAGGATCAGTTAATGGACATGAATTTGAAATTGAAGGAGAAGGAGAAGGAAGACCTTATGAAGGAACACAAACAGCAAAACTTAAAGTTACAAAAGGAGGCCCTCTTCCTTTTGCATGGGATATCCTTTCACCTCAATTTATGTATGGATCAAAAGCTTATGTTAAACATCCTGCAGATATTCCTGATTATCTTAAACTTTCATTTCCTGAAGGATTTGGCTGGGAAAGAGTTATGAATTTTGAAGATGGAGGCGTTGTAACAGTTACACAAGATTCATCTCTTCAAGATGGAGAATTCATTTATAAAGTTAAACTTAGAGGAACAAATTTTCCTTCAGATGGACCTGTTATGCAAAAAAAGACAATGGGATGGGAAGCATCATCTGAAAGAATGTATCCTGAAGATGGAGCACTTAAAGGAGAAATTAAACAAAGACTTAAACTTAAAGATGGAGGCCATTATGATGCAGAAGTTAAAACAACGTATAAAGCAAAGAAGCCTGTTCAGCTTCCTCCGGCATATAATGTTAACATTAAACTTGATATTACATCACATAATGAAGATTATACAATTGTTGAACAGTATGAAAGAGCAGAAGGAAGACATTCAACAGGAGGCATGGATGAGCTTTATAAATAA"
complete_7 = homolog_prefix + manual_7 + homolog_suffix
assert len(complete_7) == 789
complete_7.upper()



