{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a7fa8e5b-d2d8-41d9-862e-945737ca247e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8d9fa480-42c6-4e52-bae3-f476b68b86d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "sys.path.append(str(Path(os.path.dirname(os.getcwd())).parent))\n",
    "from analysis.input_testing_data.generate_input_testing_data_for_modules import generate_testing_data\n",
    "from modules.main import run_modules"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0bb16dc9-4a1c-4ebc-ba62-526c5e67b3ea",
   "metadata": {},
   "source": [
    "## Arabidopsis analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e29e5b5a-db73-4035-9a97-e2c83b51741d",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_directory_path = os.path.join(Path(os.getcwd()).parent.resolve(), \"example_data\")\n",
    "genomes_gb_path = os.path.join(data_directory_path, \"arabidopsis_microbiome\")\n",
    "zorA_file_path = os.path.join(data_directory_path, \"zorA_anti_phage_defense.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cef873b7-91a3-41a9-8ad0-cd0a1e4340cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_gb_directory = Path(genomes_gb_path)\n",
    "organisms_list = [str(f) for f in genomes_gb_directory.iterdir() if f.is_file()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cd5befe1-7d32-4802-b056-fb255024db7b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Peribacillus_frigoritolerans.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Peribacillus_frigoritolerans.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_terrigena.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_terrigena.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Terrabacter_tumescens.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Terrabacter_tumescens.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Agromyces_allii.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Agromyces_allii.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_pascens .gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_pascens .g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/File.py\", line 72, in as_handle\\n    with open(handleish, mode, **kwargs) as fp:\\nTypeError: expected str, bytes or os.PathLike object, not TextIOWrapper\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 654, in read\\n    record = next(iterator)\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/Interfaces.py\", line 72, in __next__\\n    return next(self.records)\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/GenBank/Scanner.py\", line 516, in parse_records\\n    record = self.parse(handle, do_features)\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/GenBank/Scanner.py\", line 499, in parse\\n    if self.feed(handle, consumer, do_features):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/GenBank/Scanner.py\", line 455, in feed\\n    if not self.find_start():\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/GenBank/Scanner.py\", line 88, in find_start\\n    line = self.handle.readline()\\n  File \"/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/codecs.py\", line 322, in decode\\n    (result, consumed) = self._buffer_decode(data, self.errors, final)\\nUnicodeDecodeError: \\'utf-8\\' codec can\\'t decode byte 0x80 in position 3131: invalid start byte\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/.DS_Store, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/.DS_St, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_donggukensis.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_donggukensis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "------------------------------------------\n",
      "Parsing information for Nocardioides panaciterrulae:\n",
      "------------------------------------------\n",
      "Organism is defined as wanted\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 171, in _parse_single_organism_input\\n    exp_csv_type = organism_input[\\'expression_csv_type\\']\\nKeyError: \\'expression_csv_type\\'\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_panaciterrulae.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Knoellia_subterranea.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Knoellia_subterranea.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Pedococcus_aerophilus.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Pedococcus_aerophilus.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Microcella_alkalica.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Microcella_alkalica.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Terrabacter_terrae.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Terrabacter_terrae.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Rhodanobacter_umsongensis.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Rhodanobacter_umsongensis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_aceris.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_aceris.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_silvisoli.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_silvisoli.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "------------------------------------------\n",
      "Parsing information for Peribacillus simplex:\n",
      "------------------------------------------\n",
      "Organism is defined as wanted\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 169, in _parse_single_organism_input\\n    cds = extract_gene_data(genbank_path=gb_path)\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/input_functions.py\", line 151, in extract_gene_data\\n    return extract_cds(cds_features, genome)\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/input_functions.py\", line 124, in extract_cds\\n    if any(x.gene_name == gene_name for x in cds_list):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/input_functions.py\", line 124, in <genexpr>\\n    if any(x.gene_name == gene_name for x in cds_list):\\nKeyboardInterrupt\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_ferrarius.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_ferrarius.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_parietis.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_parietis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_oryzisoli.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_oryzisoli.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Pseudarthrobacter_phenanthrenivorans.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Pseudarthrobacter_phenanthrenivorans.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_luteolus .gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_luteolus .g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_jensenii.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_jensenii.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Pedococcus_bigeumensis.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Pedococcus_bigeumensis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_silvestris.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_silvestris.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "------------------------------------------\n",
      "Parsing information for Peribacillus muralis:\n",
      "------------------------------------------\n",
      "Organism is defined as wanted\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 171, in _parse_single_organism_input\\n    exp_csv_type = organism_input[\\'expression_csv_type\\']\\nKeyError: \\'expression_csv_type\\'\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Peribacillus_muralis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "------------------------------------------\n",
      "Parsing information for Arthrobacter tumbae:\n",
      "------------------------------------------\n",
      "Organism is defined as wanted\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 171, in _parse_single_organism_input\\n    exp_csv_type = organism_input[\\'expression_csv_type\\']\\nKeyError: \\'expression_csv_type\\'\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_tumbae.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Janibacter_limosus.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Janibacter_limosus.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_alginolyticus.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Paenibacillus_alginolyticus.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "------------------------------------------\n",
      "Parsing information for Phycicoccus duodecadis:\n",
      "------------------------------------------\n",
      "Organism is defined as wanted\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 171, in _parse_single_organism_input\\n    exp_csv_type = organism_input[\\'expression_csv_type\\']\\nKeyError: \\'expression_csv_type\\'\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Phycicoccus_duodecadis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_ginsengisoli .gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Arthrobacter_ginsengisoli .g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "------------------------------------------\n",
      "Parsing information for Mycolicibacterium smegmatis:\n",
      "------------------------------------------\n",
      "Organism is defined as wanted\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 171, in _parse_single_organism_input\\n    exp_csv_type = organism_input[\\'expression_csv_type\\']\\nKeyError: \\'expression_csv_type\\'\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Mycolicibacterium_smegmatis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_baculatus.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Nocardioides_baculatus.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "------------------------------------------\n",
      "Parsing information for Phycicoccus dokdonensis:\n",
      "------------------------------------------\n",
      "Organism is defined as wanted\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 171, in _parse_single_organism_input\\n    exp_csv_type = organism_input[\\'expression_csv_type\\']\\nKeyError: \\'expression_csv_type\\'\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Pedococcus_dokdonensis.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Priestia_aryabhattai.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Priestia_aryabhattai.g, re-check your input\\n'}\n",
      "\n",
      "##########################\n",
      "# User Input #\n",
      "##########################\n",
      "Sequence to be optimized given in the following file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/zorA_anti_phage_defense.fasta\n",
      "Open reading frame sequence for optimization is:\n",
      "ATGACCGATAGCCTTAATCTTTCCAGTTTATGGCCAGACTTGAGCAGTCTAAATGTAATGCCACCTCAAACTCCTGAGCAATTGAGTGCTTTGTTTGTGGTTATTTTATGGGGTGTCGCTCTATTTTTCTTAATTTGGTCAGTTGCTGCATTTATTCGCGCAAGACAACGAGTAGTTTGGCTTAACAAATCTTTGGATGATGCGGAGAAATCGACTCTGTCAGGTGTTCGAAATGATTTAATTTCTAGAGCTGAACAAAAAAAAGATTCAGTAGGGCATTTATGGCTCGAGTTTGATGAAACATTGCTTGAAGTAAAAGGTGCTGATGACGTTGTTCGATTGCATAACACATTTGATGCTGACTATTTTTTTAATAGTTCTAGTTTAGCTGGTGGTAGTACTGAAAACAGAATGATTGCGGTGGTTCCAGGCTTTTTAACGGCTCTGGGTGTTATTGGTACTTTTGTCGGTTTACAGCTAGGACTTTCAGATCTCAATATCGCGGGCAACGTTGATGTCAATGAAATGAAAAATGGCGTTGCTGGAGTTATAAATGGTGCAAAAATCGCATTTATGACATCGGTTTGGGGCGTGCTACTCAGTGTTGCATTTAATTTCATTGAGAAGATACTTGAGCAAATTATTAGAAAGAAAATTAAGTCTTTGCAAAACCGTATTGACAGGATGTTCCCTCGATTAAGTGCGGAATATCAACTGCAATCTATTGCAAATAACAGCCAGCAATCTCGAGAAAGCCTACAAGGACTTGCTGAGAAAATAGGCGAGAAAATGCAGGAGTCACTCATTCAGGCAACTCAGGGTATTCAACAAGGTCTCGAGAGTAGTCTGGAAAAAATCATGGCTCCGGCTATCAATAAGCTCGTTGACGAAACATCTGACGGGAATCAAAAAGCACTGGAAAGCGTTCTTCAATCTTTCATGGAGGGTTTTGGTCAACAAGGCGAACAACAACGTGTCGCGATGGACTCTGCGTCTCAAAACGTAAATAAGACGCTAGAATCAATGAGCGTTTCGATGGCTGCTTTCGTCAATAAGTTGGAAGTTCAGCAAAACGCATCTTCAGAAAGAGAGAAAGAACTGATCTCAACTATTTCAGTTCAAGTTTCTGAGCTAGTTACTCAGGGTAATGAGCAGAAGCGTGTTCTAACAGAGTTTGTAGAAAAACAAGTCTCAGAGTTAGGTGAGCAGCTTACTAAAAGAGAATTAGCTGCTTCTGAACGTGAACAGAAATTGGCTTCCTCAATGGAAAGTACGATTAAAGATTTAGTAGAAAATGTTTCAGCACAAAGCCAAGTCTTGACCGATTTTGTTCAGAATCAAGTTGTCCAACTTACCCAAACGTTCTCCGAACGAGATGGAATGGCAAGTCAGATGGAAAAAGAGCGTAATGATATATTTGTGAATCAAACGCAGGCGATGAAAGCTGGAACGGATGAGCTTCTTGCTCAAGTGAAAGCGGCGACTGAATCACAACAAATCACAACTAATAGTATTATTGAGCAAGGTAAACAATTACAAAATAGTATCGATTCCTCTGTTTCGGCATCTGCTCGAGCTACGGAGAGTATGCAGCAGTCGGCGAATGAGTTACGAGTTGCCGCTGATAGTATGAATGTGTTTGGTTCCAACATCAAAGATGCGGGCAACAAACTATCTGGTGCTGTTACTGAAGCGGTTAACAGCACGAAGGATTTAGCGGAACAAAACCATCTAGGCGCAGTGAAAGTTCAAGCGTTGAGGGAGCAGTTGCTGGAAGATACAAGTAAGTTTAGCGCGATTGCAGATCAGATAAACAATATGCTGATTAGTGCAGAACAGTCGTTTAGCACGCTTCGTACTACCCAAAATGAGTTCCTTGCAGAACAAAAAGGTAATCTAAACGAACTCACTTCAGAAATGAAACGTAATGTGAAAGAGTTGACGGAGCAAATGGCTCAACTGCTTGAAGATTACGCAGAGCAAGCCAATGGACAAACCGCCGAGCACCTCAAAGTTTGGGCCAATAGTAGCACACAATATGCAGAGAGCATGAATACAGCAGTTAGAGCTATTTCTAGTGTTGTCGATGAAATTCAAGATAAAGTGAGCCAACATGCGTAG\n",
      "Encountered unknown error when running modules.\n",
      "Final output: {'error_message': 'Traceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 158, in _parse_single_organism_input\\n    gb_file = SeqIO.read(gb_path, format=\"gb\")\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/.venv/lib/python3.9/site-packages/Bio/SeqIO/__init__.py\", line 659, in read\\n    raise ValueError(\"More than one record found in handle\")\\nValueError: More than one record found in handle\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 102, in _parse_organisms_list\\n    organism = cls._parse_single_organism_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 161, in _parse_single_organism_input\\n    raise ValueError(\\nValueError: Error in genome GenBank file: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Rhodanobacter_denitrificans.gbff, make sure you inserted an undamaged .gb file containing the full genome sequence and annotations\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/main.py\", line 43, in run_modules\\n    module_input = user_IO.UserInputModule.run_module(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 30, in run_module\\n    return cls._parse_input(\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 59, in _parse_input\\n    organisms_list = cls._parse_organisms_list(organisms_input_list=module_input[\"organisms\"],\\n  File \"/Users/shimka/PycharmProjects/Igem_TAU_2021/modules/user_IO/user_input.py\", line 108, in _parse_organisms_list\\n    raise ValueError(f\"Error in organism input: {organism_key}, re-check your input\")\\nValueError: Error in organism input: /Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome/Rhodanobacter_denitrificans.g, re-check your input\\n'}\n"
     ]
    }
   ],
   "source": [
    "for first_organism in organisms_list:\n",
    "    for second_organism in organisms_list:\n",
    "        if first_organism == second_organism:\n",
    "            continue\n",
    "        input_dict = generate_testing_data(\n",
    "            orf_optimization_method = \"single_codon_diff\",  # TODO - iterate over all options here\n",
    "            orf_optimization_cub_index = \"CAI\",\n",
    "            wanted_hosts = [first_organism],\n",
    "            unwanted_hosts = [second_organism], \n",
    "            genome_path = genomes_gb_path,\n",
    "            sequence_file_path = zorA_file_path,\n",
    "            output_path = \"Arabidopsis\",\n",
    "        )\n",
    "        run_modules(input_dict, should_run_output_module=False)\n",
    "        break"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c38ec555-dfb9-42fd-b2f7-6311ac4f5018",
   "metadata": {},
   "source": [
    "### Liyam's Legacy Analysis code "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "91e402fd-4792-4532-a77f-293ef257a840",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/shimka/PycharmProjects/Igem_TAU_2021/analysis/example_data/arabidopsis_microbiome\n"
     ]
    }
   ],
   "source": [
    "final_tested_org = not_really_small_genome\n",
    "# func_options = ['single_codon_global', 'single_codon_local', 'zscore_hill_climbing_average', 'zscore_hill_climbing_weakest_link']\n",
    "func_options = ['single_codon_global', 'zscore_hill_climbing_average']\n",
    "\n",
    "spearman_dict = {}\n",
    "for translation_function in func_options:\n",
    "    opt_scores = []\n",
    "    msa_scores = []\n",
    "\n",
    "    tic = time.time()\n",
    "    for org1 in final_tested_org:\n",
    "        for org2 in final_tested_org:\n",
    "            # if org1.split(' ')[0] == org2.split(' ')[0]:\n",
    "            #     continue\n",
    "            if org1 == org2:\n",
    "                continue\n",
    "            org_dict[org1]['optimized'] = True\n",
    "            org_dict[org1]['tai_profile'] = {}\n",
    "            org_dict[org1]['tai_std'] = {}\n",
    "            org_dict[org1]['tai_avg'] = {}\n",
    "            org_dict[org2]['optimized'] = False\n",
    "            org_dict[org2]['tai_profile'] = {}\n",
    "            org_dict[org2]['tai_std'] = {}\n",
    "            org_dict[org2]['tai_avg'] = {}\n",
    "            software_dict = {\n",
    "                'sequence': cds,\n",
    "                'tuning_param': 0.5,\n",
    "                'organisms': {},\n",
    "                }\n",
    "            software_dict['organisms'][org1] = org_dict[org1]\n",
    "            software_dict['organisms'][org2] = org_dict[org2]\n",
    "            inner_tic = time.time()\n",
    "            final_cds, optimization_index, weakest_score = run_orf_optimization(software_dict)\n",
    "            print('TIME: ', time.time()-inner_tic)\n",
    "            # alignment_score = pairwise2.align.globalxx(\n",
    "            #     ribosomal_dict[org1], ribosomal_dict[org2], score_only=True)\n",
    "            # aln_scores.append(alignment_score)\n",
    "            msa_score = diff_letters(ribosomal_msa_dict[org1], ribosomal_msa_dict[org2])\n",
    "            msa_scores.append(msa_score)\n",
    "            opt_scores.append(optimization_index)\n",
    "    spearman_dict[translation_function] = spearmanr(msa_scores, opt_scores, nan_policy='omit')\n",
    "    print(spearmanr(msa_scores, opt_scores, nan_policy='omit'))\n",
    "    trans_func_name = translation_function.replace('_', ' ')\n",
    "    plt.scatter(msa_scores, opt_scores, s=0.1)\n",
    "\n",
    "            # print(org1, org2, msa_scores, optimization_index)\n",
    "    toc = time.time()\n",
    "    print(toc-tic)\n",
    "\n",
    "print(spearman_dict)\n",
    "plt.legend(['single codon optimization', 'hill climbing optimization'], loc =\"upper right\")\n",
    "plt.title(f'Evolutionary distance and model performance')\n",
    "plt.xlabel('# Different aligned characters')\n",
    "plt.ylabel('Optimization score')\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9f48742-ca32-4268-8102-96bafa1ff211",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (.venv)",
   "language": "python",
   "name": ".venv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
